{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from bs4 import BeautifulSoup as bs\n",
    "import requests\n",
    "from newspaper import Article\n",
    "import newspaper"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pyspark in c:\\users\\lalit\\anaconda3\\lib\\site-packages (3.3.0)\n",
      "Requirement already satisfied: py4j==0.10.9.5 in c:\\users\\lalit\\anaconda3\\lib\\site-packages (from pyspark) (0.10.9.5)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install pyspark"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pyspark\n",
    "from pyspark.sql import SparkSession"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "spark=SparkSession.builder.appName('pratice').getOrCreate()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Article's Title:\n",
      "Neural probe system for behavioral neuropharmacology by bi-directional wireless drug delivery and electrophysiology in socially interacting mice\n",
      "n\n",
      "Article's Text:\n",
      "Design of the wireless neural probe system integrated with a miniaturized pump\n",
      "\n",
      "The wireless neural probe system integrated with a miniaturized pump was implemented by integrating three subsystems in a single platform, i.e., (1) a neural probe embedded with microfluidic channels for the delivery of drugs and an electrode array for recording neural signals; (2) a miniaturized electrolytic pump that generates pneumatic pressure by electrolyzing the electrolyte for drug infusion; and (3) a wireless communication module that allows a user to wirelessly control the pump while simultaneously monitoring the neural activities recorded by the microelectrode array in real time (Fig. 1a). The neural probe was fabricated based on our previously reported method5,6 with modifications on the design to integrate the electrolytic pump. The neural probe largely can be divided into a body and a shank (Fig. 1b). An interdigitated electrode (IDE) for performing electrolysis and a drug inlet are located on the body of the probe. Sixteen microelectrodes, microfluidic channels, and a drug outlet are integrated on the shank of the probe (Fig. 1b). Also, the microelectrodes were electroplated with Pt black to enhance the neural recording performance17,18,19,20,21. The long-term stability of the probe, including Pt black microelectrodes, was demonstrated in our previously reported study21. In addition, the microfluidic channels were designed to have low fluidic resistance, which enables the delivery of the drug by low pressure that is generated from a miniaturized, electrolytic pump. To achieve low fluidic resistance, we maximized the number of channels to ten and increased the dimensions of the channel while maintaining the structure of the small probe (width = 10 μm; length = 7400 μm; height = 12 μm per channel). This resulted in a fluidic resistance that was four times lower than our previously reported microfluidic channels embedded in the neural probe. Thus, the width, thickness, and length of the probe shank we fabricated were 0.24, 0.04, and 6 mm, respectively (the detailed dimensions and materials of the probe were provided in Supplementary Figs. 1 and 2). The small cross-sectional area of the probe minimizes the damage to the tissue during the insertion into a brain as well as the chronic immune response in the brain7. Also, its length is sufficient to access all regions of the brain of a mouse.\n",
      "\n",
      "Fig. 1: Design and features of the wireless neural probe system with drug delivery capability. a Schematic illustrations of the wireless neural probe system and its capabilities for behavioral neuropharmacology. b Scanning electron microscope (SEM) image of the neural probe showing the interdigitated electrode (IDE) and the fluid inlet. The magnified SEM image showing a probe shank tip, including 16 microelectrodes and microfluidic channels. c Photograph showing the overall configuration of the system. d Photograph showing the wireless neural probe system mounted on the head of a mouse. Full size image\n",
      "\n",
      "The dimensions of the miniaturized electrolytic pump (width = 7 mm, length = 9 mm, and height = 6 mm) were attained by stacking three layers of polydimethylsiloxane (PDMS)-based reservoirs, including the fluidic channels, on the IDE of the body of the neural probe. The bottom reservoir contained an electrolyte (40 μl) in the bottom layer, the middle reservoir contained a drug (25 μl), and the top reservoir contained the drug that was used to refill the middle reservoir (25 μl). To infuse the drug to the outlet of the neural probe, voltage is applied to the IDE to initiate electrolysis in the bottom layer, which generates pneumatic pressure and pushes the drug in the middle layer to the outlet of the probe through the microfluidic channels (Fig. 1a). The generation of pneumatic pressure for drug infusion by electrolysis allows low power consumption and low-voltage operation without any degradation of the properties of the drug due to the generation of heat22. In addition, the electrolytic pump provides dose-controllable and chronically reliable delivery of the drug, which allows the observation of dose-dependent effects and the long-term effects of the drug.\n",
      "\n",
      "The wireless communication system is implemented with a commercial electrophysiology chip for recording and amplifying the neural signals, a microcontroller unit (MCU) for data processing, and a bi-directional Bluetooth low energy (BLE) module to control the electrolytic pump and transmit neural signals wirelessly. The BLE module provides the selection of the target animal among a crowded group as well as persistent connection even with a freely moving subject. The data processing unit (i.e., MCU) converts a large amount of analog signals recorded from the sixteen electrodes of the neural probe to only neural spikes with timing information. This allows wireless transmission of neural signals even with the low data rates (<1 Mbps) of BLE, which enables monitoring of neural activities from 16 electrodes in real time.\n",
      "\n",
      "The neural probe integrated with an electrolytic pump (0.7 g), the wireless communication system (2.36 g), and the 30 mAh Li-Po battery (0.94 g) were mounted on the head of a mouse. The components were connected electrically by flexible cables and packaged in a 3D-printed crown (0.6 g) (Fig. 1c, d). The highly packed integration of multiple modalities (i.e., drug delivery, electrophysiology, and wireless communication) in a single platform enabled the entire system to be miniaturized (~2 cm3) and have a light weight of only 4.6 g. When the system’s weight exceeds 10% of the mice’s bodyweight, it is known that their behavior could be affected23. However, after a week of recovery and adaptation to the system, we observed that there was little difference in the distance traveled by the mice before and after mounting the system (4.6 g) in the open-field test (OFT) for 30 min. Thus, we confirmed that the weight of the system did not nearly interfere with the behavior of the mice (Supplementary Figs. 3 and 4 and Supplementary Movie 1). Also, through the successive OFTs with normal mice that were not mounted with the system (Supplementary Fig. 5), we found that the average value of distance traveled was slightly decreased even though the difference was not statistically meaningful. We inferred that the slight decrease after mounting the system was due to the normal habituation to a novel environment (i.e., an open box). Thus, our system provides drug delivery and electrophysiology wirelessly in freely behaving small animals.\n",
      "\n",
      "Design and operation principle of the miniaturized electrolytic pump\n",
      "\n",
      "The electrolytic pump on the neural probe consists of three reservoir layers, two elastic membranes, and IDE on the neural probe. The reservoir layers and the elastic membranes were made of PDMS. The three reservoir layers included an electrolyte reservoir at the bottom, a drug reservoir in the middle, and a refill reservoir at the top. The two elastic membranes were placed between the layers, and the membranes separated the reservoirs and also formed check valves that allowed only unidirectional flow through the opening, which had a diameter of ~0.2 mm. The check valves in the middle of the fluidic channels are required for the electrolytic pumps to provide reliable delivery of the drug by preventing drug infusion and backflow during the resting states of the pump. Especially for in vivo applications, the absence of check valves may induce tissue suction by the backflow, causing channel clogging that results in degradation of the performance or even the failure of the device24,25. To ensure reliable drug infusion during the repeated activations of the pump, we had to eliminate the bubbles that adhered to the IDE after every operation of the pump. If the bubbles on the IDE are not removed properly, the effective area of the IDE exposed to the electrolyte would decrease, which would prevent the membrane from going to the initial position for the next injection of the drug. In addition, the pressure would build up within the enclosed electrolyte reservoir as the pump operates, and this may damage the membrane of the pump. Thus, we coated a polymer layer, i.e., Nafion (thickness = 3683 ± 257 Å), on the IDE (Supplementary Fig. 6). The recombination of gases is catalyzed by Pt IDE, and it can be accelerated by the Nafion layer, which facilitates the diffusion of gases due to its high gas solubility26. In addition to the bubbles on the IDE, the bubbles generated in the drug reservoir may cause a loss of pressure for successive drug injections. In the case of a mechanical pump, such as iPrecio which was utilized by Dagdeviren et al., it is operated by a micro-motor and may provide the accurate and reliable injection of drugs. However, there is a limitation on the miniaturization of the mechanical pump (Supplementary Table 1). Also, the noise produced during its operation may affect behavioral experiments with sensitive animals. We believe that our electrolytic pump could overcome these by miniaturizing the pump enough to be applied to small animals and by suppressing the formation of bubbles in the drug reservoir by both the refill chamber and the check value, which help provide reliable and repeatable delivery of drugs. In addition, the electrolysis-based pump can provide superior performance in power consumption and preservation of drugs compared to the heat-based pump (Supplementary Table 1). The three reservoir layers and the two elastic membranes were stacked on the body of the probe while aligning the IDE within the electrolyte reservoir and securing the flow path to the inlet (Fig. 2a). With this configuration, the miniaturized electrolytic pump operates as follows (Fig. 2b). When the pump is activated, electrolysis occurs, forming oxygen and hydrogen gases that generate pneumatic pressure and push the membrane. Then, the infusion check valve between the bottom and top layers is opened, which induces drug infusion while the refill check valve between the middle and top layers is closed, thereby blocking the fluidic channel between the drug reservoir and the refill reservoir. When the pump is deactivated, the generated gases quickly recombine into the water due to the Nafion, allowing the deflected membrane to recover, which induces backpressure to the drug reservoir from the outlet of the probe. However, the infusion check valve is closed to prevent any backflow. Also, the refill check valve is opened, which allows the drug reservoir to be refilled from the refill reservoir through the hydrophilic channel, thereby eliminating the pressure difference between the drug reservoir and the refill reservoir. Air can flow into the refill reservoir through the microscopic gaps around the plugged inlet port. Therefore, the negative pressure induced by the gas recombination can be compensated. The bubble is formed in the refill reservoir, but it cannot block the check valve because the bubbles are formed mostly from the top of the refill chamber through the gaps around the inlet port. Also, the modification of the surface by the O 2 plasma treatment during the fabrication process made it hydrophilic, thereby preventing the bubble from blocking the path of the fluid. Throughout the experiments, we activated the pump by applying pulses of 3.3 V for 5 s, which sufficiently infused the filled drug in the reservoir with low power consumption (~6 mW) (Supplementary Fig. 7 and Supplementary Movie 2; the detailed assembly method is provided in the Methods section). As described above, we successfully integrated the reliable and miniaturized pump into the neural probe. Our miniaturized pump has the unique feature of having a refill chamber with a check valve that prevents the formation of bubbles in the drug reservoir. The dimensions of our pump, including the drug volume/limits, were determined considering (1) its integration with the neural probe, (2) the alignment margins during layer-by-layer stacking, and (3) the installation on the head of a mouse. We tried to minimize the weight and size of the pump while ensuring that the pump had enough volume for chronic drug delivery.\n",
      "\n",
      "Fig. 2: Characterization of the miniaturized electrolytic pump. a Exploded view of a miniaturized electrolytic pump, consisting of three reservoir layers, two elastic membranes with inherently formed check valves, and a drug deliverable neural probe with the interdigitated electrode (IDE). b Series of schematic illustrations showing sequential on-off operations of the electrolytic pump. c Infused volume over time by activating 5 cycles with 10 s (5 s on and 5 s off), 30 s (5 s on and 25 s off), and 60 s (5 s on and 55 s off) period. The sketch inset in the graph shows the on and off times of the pump over a whole period. d Mean flow rate according to activation periods of 10 s (5 s on and 5 s off), 30 s (5 s on and 25 s off), and 60 s (5 s on and 55 s off). e Pressure required to open and close the check valves formed in the electrolytic pump (n = 5). f Normalized infusion during the 7 days of pump activation (5 cycles; 5 s on and 55 s off over a 60 s period) (n = 3). g Infused volumes according to the activation cycles of 2, 4, 6, and 8 cycles (5 s on and 55 s off over a 60 s period) (n = 4). h Normalized infusion according to the infusion media of air, agarose, and brain (3 cycles; 5 s on and 55 s off over a 60 s period) (n = 4). The sketch inset in the graphs shows the on and off times of the pump over a whole period. Data are presented as mean values ± SD with individual data points. n is the number of samples. Full size image\n",
      "\n",
      "Characterizations of the neural probe integrated with the miniaturized electrolytic pump\n",
      "\n",
      "To evaluate the performance of the neural probe integrated with the miniaturized electrolytic pump, we characterized each function of both the microelectrodes and the electrolytic pump. First, we measured the electrical impedance of the 16 microelectrodes before and after Pt black electroplating. The average impedance of the Pt electrodes before electroplating was 1.671 ± 0.104 MΩ at 1 kHz, whereas that of the Pt black electrodes after electroplating effectively was lowered to 0.016 ± 0.002 MΩ at 1 kHz. The low impedance was achieved due to the increased effective surface area, which lowers the level of the electrical noise, improves the signal-to-noise ratio, and allows efficient recording of the neural signal26,27 (Supplementary Fig. 8).\n",
      "\n",
      "Next, we characterized the drug infusion capability of the neural probe integrated with the electrolytic pump in vitro. We prepared the fabricated electrolytic pump and the neural probe to measure the infused volume with the hydraulic resistance of fluidic channels. Then, the outlet of the pump and the inlet of the fluidic channels in the neural probe were connected by a tube. While filling the reservoirs and the fluid tube with DI water, an air bubble with a volume of ~1 μl intentionally was placed within the fluid tube so that we could measure the volume of the injected fluid by the displacement of the bubble at various activation conditions. Cyclic activation of the pump was chosen because the infusion with a constant on-time was prone to tissue damage and overdosing, as well as the tearing of the membrane and bursting of the pump due to the drastic increase in the pressure28. First, we analyzed the infused volume over time by activating five cycles at various 10-s periods (5 s on and 5 s off), 30 s (5 s on and 25 s off), and 60 s (5 s on and 55 s off) (Fig. 2c). The total infused volumes were measured differently as 2.0, 1.8, and 1.7 μl for the periods of 10 s (5 s on and 5 s off), 30 s (5 s on and 25 s off), and 60 s (5 s on and 55 s off), respectively, even with the same on-time values. Thus, we measured the infusion by residual pressure during the off states for the periods of 10 s (5 s on and 5 s off), 30 s (5 s on and 25 s off), and 60 s (5 s on and 55 s off), and the corresponding infused volumes with settling time were measured as 1.1 μl for ~175 s, 0.7 μl for ~160 s, and 0.4 μl for ~120 s, respectively (Supplementary Fig. 9a). These results showed that the residual pressure might induce differences in the infusion volume, and we can control the infusion volume accurately with consideration of the residual infusion. The mean flow rates were measured as 1.15, 0.55, and 0.31 μl ∙ min−1 for the periods of 10 s (5 s on and 5 s off), 30 s (5 s on and 25 s off), and 60 s (5 s on and 55 s off), respectively (Fig. 2d). The flow rates were not higher than those of the previously reported intracerebral drug delivery system, indicating that tissue damage can be minimized irrespective of the activation period15.\n",
      "\n",
      "Also, we measured the flow rates at various on-times (3, 5, 7, and 9 s) while keeping the off-time (55 s) and the number of activation cycles (five cycles) fixed. We found that the DI water in the fluidic tube started to move from the on-time of 5 s and observed that the flow rates increased as the on-times increased (Supplementary Fig. 10a). In addition, based on the hydraulic resistance of the probe and measured flow rates, we estimated the pressure generated at various on-times (Supplementary Fig. 10b and Supplementary Note 1). The estimated pressure was 7.37 kPa at 5 s on, 15.87 kPa at 7 s on, and 18.15 kPa at 9 s on, respectively (Supplementary Fig. 10c). Thus, we confirmed that the on-time of 5 s was the minimum activation time required for initiating the infusion of the drug to the outlet of the probe (Supplementary Movie 2 and Supplementary Fig. 10), and the pressure could be controlled according to the activation time of the pump (Supplementary Fig. 10). Also, we tested the on-time of 10 s, but it was prone to the pump’s bursting and the membrane’s tearing due to the build-up of high pressure. For the in vivo experiment, we chose the on-time of 5 s and activation period of 60 s (5 s on and 55 s off) to minimize cell displacement through slow injection of the drug. Cell displacement means that cells near the electrodes are pushed temporarily by the rapid injection of the drug6,7. In particular, if the drug were to be injected rapidly, it could disturb the continuous monitoring of the neural signals from the microelectrodes after the injection of the drug6,7. During the operation of the pump, the pump demonstrated prompt initiation of drug infusion and was free of backflow due to the reliable operation of the check valve with its small opening pressure (1.22 ± 0.08 kPa) and small closing pressure (1.14 ± 0.05 kPa) (Fig. 2e).\n",
      "\n",
      "Also, we characterized the chronic reliability and dose controllability of the pump. To validate the chronic reliability of the electrolytic pump, we measured the infusion volume daily at the same operating conditions (five cycles; 5 s on and 55 s off) for 7 days to mimic the long-term investigation of the effect of the drug. During 7 days of activation with four devices, we observed that the infused volume was maintained within 7% of the initial volume (Fig. 2f). We achieved reliable drug infusion by eliminating the bubbles that were generated in both the electrolyte reservoir and the drug reservoir. Specifically, we tried to eliminate bubbles in the electrolyte reservoir by coating Nafion on Pt IDE, which promotes the recombination of the gas29. The accelerated recombination of the gas allows the rapid recovery to the initial state and enables the next electrolysis to generate a constant pneumatic pressure, acting as a reliable pumping source. Next, we integrated check valves and a refill chamber to minimize the formation of bubbles in the drug reservoir. When the pump was deactivated, the infused volume should be compensated in response to the backpressure generated by the gas recombination in the electrolytic reservoir. In the case of the normal electrolytic pump without a refill chamber and the refill check valve, the backpressure was compensated by the formation of an air bubble within the drug reservoir through gas permeable PDMS walls. Thus, the repeated operation of the normal pump resulted in enlarging the air bubble in the drug reservoir, thereby showing the degradation of the infusion performance due to the loss of pneumatic pressure by air bubble compression (Supplementary Fig. 9b). However, the proposed electrolytic pump allowed the drug reservoir to be refilled from the refill reservoir, which released the backpressure. Since backpressure was the main driving force that refilled the drug in the drug reservoir after the injection, the drug reservoir could be refilled irrespective of the amount of drug that remained in the refill reservoir, even in the later activation cycles. In addition, the refill check valve isolated the drug reservoir from the refill reservoir, thereby preventing the air bubbles in the refill reservoir from affecting the infusion characteristics. Next, we analyzed the infused volume according to the number of activation cycles to validate the controllability of the dosage. The electrolytic pump was activated at various cycles, and we found that the infused volume was linearly proportional to the number of activation cycles (R2 = 0.94), which confirmed that we could accurately control the dosage of the drug in real time (Fig. 2g). During the operation of the pump, backflow was not observed regardless of the different flow rates and infusion volumes due to the integrated check valve. Thus, we demonstrated that the injection volume could be precisely controlled by changing the activation time and the number of activation cycles without backflow.\n",
      "\n",
      "Finally, we analyzed the infusion characteristics according to the infusion medium to estimate the volume infused into the mouse brain by operating the pump with three cycles (5 s on and 55 s off). Compared with the probe shank in air, the infused volume was decreased to 65 ± 2% in agarose gel (1 wt%) and to 26 ± 5% in the brain of the mouse, respectively (Fig. 2h). We believe that the fluidic resistance, which is determined by the medium surrounding the outlet, affected the flow rate of the pressure-driven electrolytic pump. Furthermore, we monitored the infused volume for 5 days in the brain to monitor the change of the infused volume by channel clogging due to biofouling. The infused volume was kept constant under the same pump operating condition (three cycles; 5 s on and 55 s off) for 5 days (Supplementary Fig. 11). Thus, we concluded that channel clogging might not be significant and achieved reliable drug infusion even for 5 days with implantation of the probe in the brain. Throughout the characterization, we confirmed that the miniaturized electrolytic pump could deliver an accurate amount of drug directly into the brain chronically with low-voltage operation and low consumption of power. Refill can be conducted whenever necessary based on the visual inspection of the reservoirs. The animal should be anesthetized before conducting the refill process. Then, both the cured adhesive (Kwik-Sil, World Precision Instruments (WPI), USA) and the plug in the inlet port of the reservoir are removed carefully. After using a syringe to refill the chamber, we plug the reservoir and cast adhesive again.\n",
      "\n",
      "Design and operation principle of the wireless bi-directional communication module\n",
      "\n",
      "The bi-directional wireless communication module is composed largely of two parts, i.e., (1) a signal recording part for sending neural signals and (2) a pump control part for receiving signals for the operation of the pump (Supplementary Fig. 12a). We used dedicated Bluetooth communication modules for each part to enable the simultaneous operations of the neural signal recording through a PC and the pump control through a smartphone (Supplementary Fig. 12b). Also, the overall system was designed to be light-weighted by integrating electrical components on a thin 0.5 mm printed circuit board (PCB). The signal recording part consists of a signal amplifier chip (RHD2132, Intan Technologies, USA) that amplifies the recorded neural signals from the 16 microelectrode array, an MCU (TI MSP 430, Texas Instruments, USA) for processing recorded signals, and a Bluetooth transceiver (PAN1326B, Panasonic Electronic Components, USA) for wireless transmission of the processed signal data to a Bluetooth receiver connected to the PC (Supplementary Fig. 12b). We minimized the power consumption by adopting low-power electronic components. Also, to overcome the limitation of the number of measurable channels due to the low bandwidth (<1 Mbps) of the Bluetooth communication protocol, we applied a real-time spike detection protocol in the system30. By only transmitting neural spikes above the noise level (~50 μV) with timing information by the MCU, we were able to measure neural signals from 16 electrodes simultaneously.\n",
      "\n",
      "The pump control part consists of a BLE module embedded with MCU (FBL770BC, Firmtech, South Korea) and a p-channel MOSFET for a stable power supply during the operation of the pump. We can control the pump using an Android-based smartphone through the application provided by the manufacturer (Supplementary Fig. 12b). Thus, we were able to initiate or stop infusion and wirelessly control the dose of the drug that was delivered at any time. The power consumption of the bi-directional communication module was 74 mW during the transmission of the recorded neural signal and 5 mW for the wireless pump control (Supplementary Table 2).\n",
      "\n",
      "The weight and size of the bi-directional wireless module were 2.36 g and 1350 mm3, respectively, which was sufficiently small to be applied to the head of a mouse31 (Supplementary Table 2). In addition, due to the low consumption of power, the system could operate for about an hour using a 30 mAh lithium polymer (Li-Po) battery that weighed only 0.94 g. The weight of the proposed system is slightly higher than wireless systems that only have stimulation capability, but the weight is similar to that of the previous wireless recording system even though the proposed system has an additional wireless pump module (Supplementary Table 2). Also, our system available could be applied to social experiments that require the simultaneous recording of the neural signal among multiple mice and the delivery of the drug using the bi-directional wireless module based on the BLE protocol that minimizes signal interference among wireless systems.\n",
      "\n",
      "Wireless drug delivery and simultaneous monitoring of neural and behavioral changes\n",
      "\n",
      "The key advantage of our system is its ability to perform wireless drug delivery, electrophysiology, and behavior monitoring simultaneously. To demonstrate the simultaneous monitoring of both behavior and neural signals by the drug that was delivered, we implanted the neural probe in the substantia nigra (SN) of C57Bl6 wild-type mice and delivered bicuculline (BIC), a GABA A receptor antagonist, into the targeted region in the brain (detailed surgical procedure are provided in the Method section, Supplementary Fig. 13, and Supplementary Movie 3). It is well known from previous studies that injection of BIC into the SN induces both an increase in neural activities and compulsive circling behavior32,33. First, we connected a flexible cable to FFC/FPC connectors of the wireless module (Supplementary Movie 4). The flexible cable was connected easily with a little force, because the other end of the flexible cable was rigidly fixed to the probe’s PCB by epoxy. (The detailed process for packaging the probe is provided in the Methods section.) Then, we connected the battery to the power connector of the wireless module (Supplementary Movie 4). For the real-time observation of the effects of the drug, we performed an OFT without drug infusion for 10 min as a control and activated the electrolytic pump wirelessly for three cycles (3.3 V; 5 s on and 55 s off) and infused ~300 nL of 0.24 mM BIC into SN (Fig. 3a). During the injection of BIC, the animal showed a little behavioral change, such as freezing (Supplementary Movies 5 and 6). The operation of the pump or the liquid infusion might have affected the behavior. However, previous research showed that the aCSF (artificial cerebrospinal fluid) infusion did not affect behavioral change16. Also, the electrolyte pump does not produce any sound that can be generated by a piezoelectric pump (Supplementary Table 1). Thus, we inferred that the behavioral change during the injection of BIC might have been due to some other reason, and this will be investigated through further studies. After the injection of BIC, we observed the change in the behavior of the mice from exploratory behavior to compulsive circling (Fig. 3b and Supplementary Movies 5 and 6). Also, Through the real-time monitoring of the neural activities, the firing rates of SN neurons were increased after the injection of BIC (Fig. 3c, d and Supplementary Movies 5 and 6). Specifically, the firing rates started to be increased (more than 150% firing rate compared to that before injection) about 1 min after the pump was activated (Supplementary Fig. 14). As the infused volume increased by three on-off cycles of the pump, the firing rates increased gradually. After the injection of BIC was finished, the increased firing rates gradually returned to the rates before the injection of BIC due to the dilution of the BIC in the brain (Supplementary Fig. 14).\n",
      "\n",
      "Fig. 3: Wireless drug delivery and simultaneous monitoring of neural, behavioral changes. a Series of schematic illustrations showing the behavioral change induced by the injection of bicuculline (BIC) into the substantia nigra (SN) and experimental timeline. b Series of heatmaps showing the behaviors of a mouse before and after the BIC injection (5 min period). c Real-time circling behavior and neural activities of SN neurons and merged firing rate of SN neurons in accordance with BIC injection. The blue line above the heatmap indicates the time at the pump on (3 cycles; 5 s on and 55 s off over a 60 s period). The pink line above the heatmap indicates the time at which the circling behavior appeared. The inset shows representative neural signals. d Neural firing patterns of SN neurons before and after BIC injection as well as after recovery. “After recovery” was based on the time when the compulsive circling behavior changed to normal behavior (e.g., walking behavior). e The number of circling per minute before and after BIC injection, and after recovery (n = 9 where n is the number of repetitive experiments with the 3 mice (i.e., three experiments per mouse); F(1.193, 9.547) = 63.64, p = 0.00001029). f Mean firing rates of SN neurons that showed firing pattern change before and after BIC injection as well as after recovery (n = 8 where n is the number of neurons from 3 mice; F(1.442, 10.09) = 12.31, p = 0.00326). g The number of burst events for 30 s of some SN neurons before and after BIC injection as well as after recovery (n = 8 where n is the number of neurons from 3 mice; F(1.074, 7.520) = 13.31, p = 0.00670). Data are presented as mean values ± SD with individual data points. Statistical analyses were performed by one-way repeated measures ANOVA with Tukey’s multiple comparisons test. p < 0.05 was considered significant. *p < 0.05, ***p < 0.001, ****p < 0.0001. ns no statistical significance. Full size image\n",
      "\n",
      "We analyzed the changes in the circling behaviors quantitatively, and we analyzed the firing rates of the neural signals before and after the injection of BIC. We observed an increase by approximately a factor of 4 when we compared the number of circling behaviors before and after the injection of BIC (Fig. 3e). Also, after the BIC injection, the mean firing rate increased by a factor of 2 (Fig. 3f). The increase in the firing rate came from the change of firing patterns changing from spontaneous firing to burst firing, which was recovered afterward (Fig. 3d, f and Supplementary Movie 6)34. After the injection of BIC, the burst rate was increased significantly, resulting in an increased firing rate, and, later, the changes returned to the baseline level (Fig. 3g). Their behavior also slowly returned from compulsive circling behavior to normal behavior (Supplementary Movie 6). From the in vivo experiment using wireless BIC infusion, we showed that our wireless neural probe system could deliver a drug while monitoring the changes in behavior and neural activities, including the real-time firing patterns.\n",
      "\n",
      "Controllable, chronic drug delivery and consequent changes in neural, behavioral activities\n",
      "\n",
      "Next, we evaluated the chronic reliability and dose controllability of the electrolytic pump of the proposed system. To demonstrate the long-term reliable delivery of the drug, we injected BIC into the SN of a mouse daily for a week using the same pump activation condition described in the previous section. We monitored the neural activity and behavioral changes in real-time according to the daily BIC injection (Fig. 4a, b and Supplementary Movie 7). The number of occurrences of circling behavior as well as the mean firing rate was increased significantly after the BIC injection, and the changes induced by the BIC were consistent for 7 days (Fig. 4c, d).\n",
      "\n",
      "Fig. 4: Controllable, chronic drug delivery, and consequent neural and behavioral changes. a Firing rates of a representative SN neuron in accordance with BIC injection on day 3, day 5, and day 7. Inset shows a representative neural signal. The blue line above the heatmap indicates the time at the pump on (3 cycles; 5 s on and 55 s off over a 60 s period). b Behavior heatmap of a representative mouse in accordance with BIC injection on day 3, day 5, day 7 (5 min period). c Number of circling before and after BIC injection daily for 7 days (5 min period; n = 3 where n is the number of mice). d Increase in the mean firing rates after the injection of BIC daily for 7 days (n = 4 where n is the number of neurons). e Representative trajectories with velocity heatmap after BIC injection by activating the pump with 3 cycles and 6 cycles (5 min period). f The number of circling after BIC injection by activating the pump with 0 cycle, 3 cycles, and 6 cycles (5 min period; n = 8 where n is the number of repetitive experiments; F(2, 21) = 30.25, p = 0.00000065). g Travel distance after BIC injection by activating the pump with 0 cycle, 3 cycles, and 6 cycles (5 min period; n = 8 where n is the number of repetitive experiments; F(2, 21) = 87.11, p = 0.000000000068). Data are presented as mean values ± SD with individual data points. Statistical analyses were performed by the one-way ANOVA with Tukey’s multiple comparisons test. p < 0.05 was considered significant. **p < 0.01, ***p < 0.001, ****p < 0.0001. ns no statistical significance. Full size image\n",
      "\n",
      "To demonstrate the dose-controllable drug delivery, we injected a different amount of BIC by activating the electrolytic pump with three and six cycles, respectively. Both conditions induced circling behavior, but the mouse injected with the higher BIC dose (i.e., six-cycle activation) showed more intense circling behavior, rotating with larger circles and higher velocities (Fig. 4e and Supplementary Movie 8). Then, we quantitatively analyzed and compared the behavioral responses to control (no BIC injection), low BIC dose (three-cycle activation), and high BIC dose (six-cycle activation). Consequently, we observed that the number of circling behaviors was proportional to the activation cycle (Fig. 4f). We also observed different travel distances depending on the dosage of the injected BIC (Fig. 4g). Interestingly, the low BIC dose resulted in the least travel distance, and it was even less than the control case. This was due to the relatively slow, somewhat stiffened, circling behavior compared to the freely behaving exploration shown in the control cases. However, the high BIC dose showed a drastic increase in the travel distance due to the intense circling behavior. Through these in vivo experiments, we successfully demonstrated that our system could be utilized in analyzing chronic, dose-dependent drug effects through the real-time monitoring of the changes in the behavioral aspects and the changes in the neural signals of freely behaving mice.\n",
      "\n",
      "Modulation of feeding behavior by drug delivery and simultaneous monitoring of changes in the neural signals\n",
      "\n",
      "The wireless drug delivery capability of our system also allows us to modulate the behavior of an animal in natural status, which also can be correlated with the neural response. This feature potentiates the neuropharmacological study of physiological behaviors that are involved in complex neural circuits and require multi-lateral validation. In this experiment, we applied the system to modulate the feeding behavior of freely behaving mice by delivering the drug to the region of the brain related to feeding and correlate with the change in their neural activities. Previous studies have reported that GABAergic neurons in the lateral hypothalamus (LH) manage satiety, and a GABA A receptor agonist, muscimol, modulates the neural circuit related to the feeding behavior35. To modulate feeding behavior, we implanted the neural probe system in the LH of wild mice followed by restricted caloric intake for 1 week after recovery. A caloric-restricted mouse was placed in a home cage for 20 min with a food pellet in the center, and the behavior and food intake of the mouse were compared according to the muscimol (1 mM) injection (Fig. 5a, Supplementary Fig. 15, and Supplementary Movie 9). In the case of the mouse without the muscimol injection, feeding behaviors were frequently observed, which resulted in the concentrated nose position around the food pellet (Fig. 5b). However, when muscimol was injected into the LH of the same mouse by activating the electrolytic pump for three cycles (~300 nL delivery), feeding behavior was highly suppressed, and the mouse showed reduced feeding behaviors with the nose position far from the food pellet (Fig. 5c). Throughout the experiments, muscimol injection resulted in the reduced time spent around the food pellet, reduced feeding duration, and reduced consumption of food pellets, clearly indicating the suppressed intake of food (Fig. 5d–f). We also performed real-time monitoring of the LH neurons after the injection of muscimol (Fig. 5g). While the mean firing rate of the LH neurons did not show a significant change before the muscimol injection, we observed an approximately three-fold decrease in the mean firing rate after the muscimol injection (Fig. 5h). This experiment clearly indicated that our system could be applied to the drug-induced neuromodulation with monitoring of the corresponding behavior and electrophysiological change.\n",
      "\n",
      "Fig. 5: Modulation of feeding by drug delivery and simultaneous monitoring of neural changes. a Schematic illustration showing the experimental setup for food intake test. b, c Behavior heatmap expressed based on nose positions in b a control experiment and c muscimol-injected experiment. The blue line above the heatmap indicates the time at the pump on (3 cycles; 5 s on and 55 s off over a 60 s period). d Comparison of time spent around the food pellet between the control and muscimol-injected experiments (n = 6 where n is the number of mice; p = 0.03125). e Comparison of feeding duration between control and muscimol-injected experiments (n = 6 where n is the number of mice; p = 0.03125). f Comparison of food intake between the control and muscimol-injected experiments (n = 6 where n is the number of mice; p = 0.03125). g Real-time neural activities of LH neurons and merged firing rate of LH neurons in accordance with the injection of 1 mM of muscimol. The blue line above the heatmap indicates the time at the pump on (3 cycles; 5 s on and 55 s off). The inset shows representative neural signals. h Change in the mean firing rate of LH neurons depending on the injection of muscimol (n = 9 where n is the number of neurons; p = 0.00391). Data are presented as mean values ± SD with individual data points. All statistical analyses were performed by the Wilcoxon matched-pairs signed rank two-tailed test. p < 0.05 was considered significant. *p < 0.05, **p < 0.01. Full size image\n",
      "\n",
      "Measurement of neural activity of two mice in competition for food\n",
      "\n",
      "The significant advantage of a wireless system is the ability to monitor the behaviors of multiple animals without any physical restrictions, which allows the precise observation of social interactions. With wireless drug delivery, we can even modulate socially interacting behaviors between mice. We designed an experiment that effectively showed the modulation of social behavior by drug delivery to the brain. During the competition for food, two mice endeavor to have the same food. However, as shown in the previous section, it is possible to modulate feeding behavior by delivering muscimol to LH. Therefore, we expected that the modulation of feeding behavior on one mouse by drug injection could also affect the social interaction (i.e., from competition to non-competition for food) between two mice during food competition.\n",
      "\n",
      "During the competition for food prior to the neuromodulation experiment, we measured the neural activities of a pair of freely behaving mice during their competition for food. Previous studies have reported that the medial prefrontal cortex (mPFC) neurons are related to general social behaviors, including recognition36,37, interaction38,39, and particularly competition19,40. Thus, we implanted a neural probe in the mPFC region of a mouse with a green LED attached (i.e., green mouse) to record the neural signal related to social behavior. Next, we implanted a neural probe with drug delivery in the LH region of the mouse with a red LED attached (i.e., red mouse) to both record the neural signal and modulate the feeding behavior. Before the competition for food, both mice experienced caloric restriction to induce proper food competition between them. Also, the mice were trained individually for a week to adapt to the food provision protocol. (The detailed preparation procedures for the competition for food are provided in the Methods section.)\n",
      "\n",
      "The food competition was performed in a custom-made activity chamber that measured 40 × 12.5 × 20 cm3 and consisted of a port for providing food and a green LED as a visual cue for noticing the provision of food (Fig. 6a). Half of the chamber away from the port where the food was provided was set as a start zone. When both mice were positioned in the start zone, a bite-sized food pellet (20 mg) was provided along with the visual cue, and then the competition began (Fig. 6b and Supplementary Fig. 16). The food pellet was provided only once in a trial, and food was provided only when both mice stayed in the start zone during the entire trial period, which lasted 2 min. (The detailed method and procedure in the food competition test are provided in the Methods section.)\n",
      "\n",
      "Fig. 6: Food competition between two freely behaving mice and real-time monitoring of the neural activities. a Schematic illustrations showing the experimental setup and a brief protocol for the food competition test. b Sequential photographs captured during a food competition test, showing (i) food provision, (ii) food preoccupancy by one mouse, and (iii) food competition. c Neural activities of a green mouse (mPFC neurons) and a red mouse (LH neurons) in accordance with competition and non-competition trials during a representative food competition test. d Real-time firing rate of mPFC neurons and e LH neurons in accordance with behavioral events during a representative food competition trial. Inset shows representative neural signals. f Mean firing rate of LH neurons (n = 19 where n is the number of LH neurons; p = 0.83996) and g mPFC neurons before and after proving the food pellet (n = 40 where n is the number of mPFC neurons; t(78) = 10.89, p < 0.000000000000001). h Overall winning history of three mouse pairs during four food competition tests (120 trials). i Overall competition occurrence within the non-passed trials (111 trials). j Number of winning per test for each mouse (n = 12 where n is the total number of games with 3 pairs of mice; t(11) = 1.493, p = 0.16367). Data are presented as mean values ± SD with individual data points. Statistical analyses were performed by the two-tailed Mann–Whitney test in f, the two-tailed unpaired t-test in g, and the two-tailed paired t-test in j. p < 0.05 was considered significant. ****p < 0.0001. ns no statistical significance. Full size image\n",
      "\n",
      "First, we analyzed the real-time neural activities of each mouse during a food competition test (Fig. 6c and Supplementary Movie 10). During a 20-min competition test, the green mouse showed an increase in the firing rates of mPFC neurons after providing the food pellet, but no significant change in the firing rates of LH neurons was observed from the red mouse during the same competition test. In the case of the green mouse, the mPFC neurons were activated continuously, from the moment the mouse saw the food pellet until the end of the competition (Fig. 6d). Specifically, the firing rate of the neural spike started to increase as soon as the green mouse noticed the visual cue and started to run for the food. Also, increased firing rates of the mPFC neurons were maintained when the opponent approached and guarded the opponent’s food pellets. Based on these results, we could infer that the mPFC neurons are involved extensively in social36,37,38,39 and non-social behaviors19,41, such as approach, competition, recognition, and seeking rewards. In the case of the red mouse, the LH neurons did not show any significant change during the same competition trial, indicating that the LH region was not related to competitive behaviors (Fig. 6e). These results were consistent throughout the experiments, and they were confirmed statistically (Fig. 6f, g and Supplementary Fig. 17a).\n",
      "\n",
      "Then, we statistically analyzed the results of the food competition. Three pairs of mice underwent 120 competition trials (3 mouse pairs × 4 competition tests × 10 trials per test). Throughout the competitions tests, the red mice showed a 37.5% (45/120) winning rate, while the green mice had a winning rate of 54.2% (65/120) winning rate, and the remaining 7.5% (9/120) were classified as passed trials (Fig. 6h and Supplementary Fig. 17b). Within the non-passed trials, the competition occurred at a rate of 79.3% (88/111), while a 20.7% (23/111) rate was observed for the one-sided acquisition of food without competition (Fig. 6i and Supplementary Fig. 17b). In summary, the green mice won more times, but there was no quantitative difference between the number of times each mouse won the competition, which indicated that it was not a one-sided winning (Fig. 6j and Supplementary Fig. 18). Overall, we demonstrated that the food competition protocol did not show one-sided winning, and we were able to measure the neural signals successfully during the competition and to verify the correlation of LH neurons and mPFC neurons with the food competition.\n",
      "\n",
      "Modulation of the food competition between two mice by the delivery of the drug\n",
      "\n",
      "After we confirmed the food competition protocol, we demonstrated the modulation of feeding behavior by the wireless drug delivery to one mouse during the competition test to further highlight the neuropharmacological utility of our system for studying social interactions. From the analysis of the neural signals and the behavioral responses to the in vivo drug delivery, we obtained interesting results, i.e., results that were not feasible by the existing neural interfaces. To modulate feeding behavior in the food competition, we injected muscimol (1 mM; ~300 nL injection by three-cycle activation) into the LH of the red mouse and monitored the changes in neural activities as well as the social interaction (i.e., competition) (Supplementary Fig. 19). During the competition test with the muscimol injection, the mPFC neurons of the green mouse showed an increased mean firing rate during the competition compared to the rate before the competition, as shown in the results in the previous section (Fig. 7a). However, the mean firing rate of the LH neurons of the red mouse decreased after the injection of muscimol (Fig. 7a).\n",
      "\n",
      "Fig. 7: Drug-induced modulation of the food competition and simultaneous monitoring of the neural and social interaction changes. a Neural activities of a green mouse (mPFC neurons) and a red mouse (LH neurons) in accordance with competition patterns during a representative food competition test with muscimol injection into the LH of the red mouse. The blue line above the heatmap indicates the time at the pump on (3 cycles; 5 s on and 55 s off over a 60 s period). b, c Real-time firing rate of mPFC neurons and LH neurons in accordance with behavioral events during a representative food competition trial b before and c after the muscimol injection. d Representative photographs taken before and after the muscimol injection, showing the absence of competition after the muscimol injection. e Overall winning history and competition occurrence before, during, and after the muscimol injection during four food competition tests conducted with three pairs of mice. f Mean firing rate of LH neurons (n = 19 where n is the number of LH neurons; before: p = 0.72911; after: p = 0.13197) and g mPFC neurons before and after providing the food pellet depending on the muscimol injection (n = 40 where n is the number of mPFC neurons; before: t(78) = 9.762, p = 0.000000003; after: t(78) = 3.485, p = 0.00081). h Mean firing rates of mPFC neurons during the competition depending on the muscimol injection (n = 40 where n is the number of mPFC neurons; t(78) = 6.691, p = 0.000000003). Data are presented as mean values ± SD with individual data points. Statistical analyses were performed by the two-tailed Mann–Whitney test in f and the two-tailed unpaired t-test in g and h. p < 0.05 was considered significant. ***p < 0.001, ****p < 0.0001. ns no statistical significance. Full size image\n",
      "\n",
      "However, after the injection of muscimol into the LH of the red mouse, we found interesting results, i.e., the firing rate of the mPFC neurons increased only after the food pellet was provided (Fig. 7c), while the mPFC neurons showed increased activity during the entire period of the competition before the injection of muscimol (Fig. 7b). The change in the mPFC neurons by the injection of muscimol into the opponent mouse can be explained by the change in social interaction. Before the injection of muscimol, we were able to observe the social interaction between them, such as the approach to the opponent mouse and the competition for the food. However, after the injection of muscimol into the LH of the red mouse, its appetite was suppressed, and their social interaction did not occur (Fig. 7d, Supplementary Fig. 20, and Supplementary Movie 11), which might induce the decrease in mPFC activity after muscimol injection.\n",
      "\n",
      "Next, we statistically analyzed the winning rates of the two mice according to the muscimol injection (Fig. 7e and Supplementary Figs. 20 and 21). Within the 120 competition trials (3 mouse pairs × 4 competition tests × 10 trials per test), 48 trials were performed before or during the injection of the muscimol, and 72 trials were performed after the injection of the muscimol. Before and during the injection of muscimol, two mice showed similar winning rates, i.e., 50% (24/48) for the red mouse and 41.7% (20/48) for the green mouse. However, after the injection of muscimol, the green mouse won all of the competition except for the passed trials. Moreover, 100% of the non-pass trials were non-competition trials, which indicated that the green mouse acquired all of the provided food without any competition after the injection of muscimol into the red mouse (Supplementary Figs. 20 and 21). These results strongly indicated that the injection of muscimol into the LH successfully suppressed the appetite of the red mouse.\n",
      "\n",
      "In addition, we statistically analyzed the changes in the neural activity in LH and mPFC according to the muscimol injection. As expected, the mean firing rate of LH neurons was decreased, and that of the mPFC neurons was still increased after the muscimol injection (Fig. 7f, g and Supplementary Fig. 22a–f). However, we found a significant decrease in the mean firing rate of the mPFC neurons (Fig. 7h, Supplementary Figs. 22g–I and 23–25) because there was no social interaction, including competition for the food pellet, after the injection of muscimol. From the food competition experiments, we successfully modulated feeding behavior, i.e., by administering the drug to one mouse, we induced a change in the social interaction, including competition for the food, and we observed changes in the neural activities of both mice, neither of which has ever been demonstrated before.\n",
      "\n",
      "Change in the neural activities of mPFC by successive non-participation of the competitor during food competition test\n",
      "\n",
      "When we further analyzed the firing rate of the mPFC neuron in the green mouse after injecting muscimol into the LH of the red mouse, we found an interesting result, i.e., a change in neural activity. After the injection of muscimol into the LH of the red mouse, the mouse no longer participated in the competition for food, and the green mouse took a food pellet without competition in 6 successive trials (Figs. 7e and 8a, b). Interestingly, the neural activities of mPFC decreased gradually as the number of trials increased despite the fact that the green mouse was experiencing the same non-competitive trials (Supplementary Fig. 26 and Supplementary Movie 12). In all six successive trials, the neural activity of mPFC neurons increased when the green mouse recognized the food pellet and ran to eat it, as with the above result and previous studies19,41. However, the mPFC neurons during the first trial after the injection of muscimol showed longer and higher neural activity than those during the sixth trial (Fig. 8c). We also analyzed the mean firing rates of the mPFC neurons for the six successive non-competition trials to investigate changes in the mPFC neurons as the trials proceeded (Fig. 8d and Supplementary Fig. 27). The mean firing rate initially was high for the first two trials, but after the muscimol injection, the mean firing rate decreased gradually until the fourth trial, and it remained low afterward. The gradual decrease of the mean firing rate in the same non-competition trials may indicate that (1) as sessions without competition continued, the green mouse no longer regarded these sessions as competition, (2) the effort for learning the competition rule related to green LED was changed over time, or (3) social interaction was decreased. In addition, we analyzed the time for the green mouse to reach the food pellet during successive, non-competition trials. The average time to reach the food pellet increased slightly over the trials, but there was no statistical difference between the trials (Supplementary Fig. 28). This result could be inferred as the change of motivation for the food.\n",
      "\n",
      "Fig. 8: Change in the neural activities of mPFC by successive non-participation in competition of the competitor during food competition test. a Schematic illustrations showing the unilateral food winning by the green mouse without the food competition in the successive 6 trials after muscimol injection into the LH of the red mouse. b Sequential photographs captured during non-food competition trial after muscimol injection into the LH of the red mouse. c Dynamic changes of mPFC neural activities in the green mouse during the first and sixth trials after the injection of muscimol into the LH of the red mouse. d Comparison of mPFC neural activities between trials after the injection of muscimol into the LH of the red mouse (n = 40 where n is the number of mPFC neurons; F(2.579, 100.6) = 29.89, p = 0.000000000001029). Data are presented as mean values ± SD with individual data points. Statistical analysis was performed by the one-way repeated measures ANOVA with Tukey’s multiple comparisons test. p < 0.05 was considered significant. **p < 0.01, ***p < 0.001. ns no statistical significance. Full size image\n",
      "\n",
      "This in-depth analysis that correlates the activities of the mPFC neurons with social interest against the competitor or the changes of perception against the test or the decrease in social interaction was possible due to the ability to simultaneously observe the neural activities and behaviors in real time while modulating the food competition by injecting the drug into the competitor. Throughout the food-competition experiments, we demonstrated that, by using our system, even the experiments for studying the correlation between the brain and social behaviors caused by the drugs could be performed sufficiently with a high degree of freedom.\n",
      "n\n",
      "Article's Summary:\n",
      "Also, through the successive OFTs with normal mice that were not mounted with the system (Supplementary Fig.\n",
      "c Infused volume over time by activating 5 cycles with 10 s (5 s on and 5 s off), 30 s (5 s on and 25 s off), and 60 s (5 s on and 55 s off) period.\n",
      "d Mean flow rate according to activation periods of 10 s (5 s on and 5 s off), 30 s (5 s on and 25 s off), and 60 s (5 s on and 55 s off).\n",
      "Before the competition for food, both mice experienced caloric restriction to induce proper food competition between them.\n",
      "b Sequential photographs captured during a food competition test, showing (i) food provision, (ii) food preoccupancy by one mouse, and (iii) food competition.\n",
      "n\n"
     ]
    }
   ],
   "source": [
    "url1 =\"https://www.nature.com/articles/s41467-022-33296-8\"\n",
    "toi_article = Article(url1, language=\"en\")\n",
    "toi_article.download()\n",
    "toi_article.parse()\n",
    "toi_article.nlp()\n",
    "print(\"Article's Title:\")\n",
    "print(toi_article.title)\n",
    "print(\"n\")\n",
    "print(\"Article's Text:\")\n",
    "print(toi_article.text)\n",
    "print(\"n\")\n",
    "print(\"Article's Summary:\")\n",
    "print(toi_article.summary)\n",
    "print(\"n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "#pip install newspaper3k"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Article's Title:\n",
      "Reaction-based fluorogenic probes for detecting protein cysteine oxidation in living cells\n",
      "n\n",
      "Article's Text:\n",
      "General spectroscopy\n",
      "\n",
      "Fluorescence spectra for chemical and biochemical studies were recorded on an Agilent Cary Eclipse Fluorescence Spectrophotometer and analyzed using Eclipse software v1.2. Absorbance spectra were recorded on an Agilent Cary 300 Bio UV-Visible Spectrophotometer and analzyed using WinUV software v4.20.\n",
      "\n",
      "Kinetic studies\n",
      "\n",
      "Stock solutions of compounds 1–8 were prepared in DMSO at a final concentration of 100 mM and then diluted in PBS. A stock solution of dipeptide sulfenic acid (CSA)19 was prepared in acetonitrile at a final concentration of 100 mM. To a solution of compound (2 mL), was added a solution of CSA (1 mL) and incubated at room temperature. Final concentrations were as follows: Compounds 1–8: 25, 100, or 250 µM; CSA: 2.5, 10, or 25 µM adhering to pseudo first-order conditions. Aliquots (300 µL) of the reactions were collected at regular intervals and quenched by the addition of formic acid (100 µL). The resulting samples were analyzed LC-MS using an Agilent 1220 Infinity LC system coupled to a 6120 Quadrupole running OpenLAB CDS software vA.03.02.024. The amount of product plotted versus time to obtain k obs using KaleidaGraph software v 4.1.1. Reactions of 7 or 8 were also measured at 10 µM final concentration and CSA at 100-500 µM final concentration in PBS:ACN (9:1) pH = 7.4 were prepared and the fluorescence signal was immediately recorded for 5-10 min. Fluorescence versus time was plotted to obtain k obs which were plotted against the concentration of CSA to obtain second-order rates constants.\n",
      "\n",
      "Fluorescence quantum yield (ϕ) calculation\n",
      "\n",
      "Lucifer Yellow (LY, φ LY = 0.21 in water) was used as a general reference for all analytes due to proximity of its excitation λ max (428 nm) to those from unreacted CysOx1 or CysOx2 and their covalent CSA adducts (443–447 nm). Three dilute solutions of each compound with absorbances between 0.01 and 0.1 at the λ max were prepared in PBS. Emission spectra of these solutions using each analyte’s λ max as excitation wavelength were recorded and areas under the emission spectra were integrated. Emission spectra of the LY reference were obtained using different excitation wavelengths corresponding to each analyte’s λ max to ensure the same excitation photon density between reference and analyte. Linear regression of area of fluorescence versus absorbance provided slopes (m), which were applied in the following equation to calculate φ for each analyte, where the subscript LY refers to data obtained with Lucifer Yellow:\n",
      "\n",
      "$$\\phi={\\phi }_{{{{{{\\rm{LY}}}}}}} \\times \\frac{m}{{m}_{{{{{{\\rm{LY}}}}}}}}$$\n",
      "\n",
      "Intact mass spectrometry (MS) of C64,82S Gpx3\n",
      "\n",
      "C64,82S Gpx3 was thawed at 0 °C and reduced with 50 mM (2 M, 5 µL) DTT for 20 min on ice. C64,82 S Gpx3-SH was exchanged into labeling buffer (50 mM HEPES, 100 mM NaCl, pH = 7.4) using P-30 spin columns (Bio-Rad). The C64,82 S Gpx3-SH solution was diluted in labeling buffer to give a final protein concentration of 10 µM in a reaction volume of 200 µL. 8 µL of a 25 mM DMSO solution of CysOx1 or CysOx2 was added, followed by 3.0 µL of 1 mM H 2 O 2 or 700 mM TCEP in ddH 2 O. Reaction mixtures were incubated at room temperature for 1 h, spin filtered using P-30 spin columns, and resolved on a C4 column (Grace Vydac, Cat. No. 214TP3405) connected to an Agilent 1220 Infinity LC. The molecular mass of CysOx-labeled C64,82 S Gpx3 was measured on a Thermo Scientific LTQ XL linear Ion trap mass spectrometer. Spectra were acquired in the positive ion mode and deconvoluted using MagTran software v1.0.\n",
      "\n",
      "In situ reaction of C64,82S Gpx3 with CysOx1 or CysOx2\n",
      "\n",
      "Solutions of C64,82S Gpx3-SH (20 µM) and CysOx1 or CysOx2 (20 µM) were prepared in labeling buffer and stored on ice. The Cary Eclipse software was set to Kinetics mode using the fluorescence excitation and emission maxima observed for Gpx3-CysOx adducts (357/498 nm for Gpx3-CysOx1 and 394/535 nm for Gpx3-CysOx2). C64,82 S Gpx3-SH (50 µL) and CysOx probe (50 µL) were mixed in a cuvette. Next, H 2 O 2 (100 µL of 30 µM in labeling buffer) was added and fluorescence measurements were acquired in triplicate.\n",
      "\n",
      "In-gel fluorescence analysis of CysOx-labeled proteins\n",
      "\n",
      "Gpx3 labeled with CysOx1 or CysOx2 was diluted in reducing SDS-PAGE load dye, boiled for 5 min, and stored on ice. For HeLa cell lysates derived from CysOx-treated cells 20 µg of protein was diluted to a final volume of 30 µL in non-reducing load dye, boiled for 5 min, and stored on ice. SDS-PAGE precast 4–15% or 4–20% gradient gels (Bio-Rad) were electrophoresed in Tris-Glycine-SDS running buffer at 80 V for 10 min and then at 120 V for 60 min. Gels were then rinsed with ddH 2 O and imaged using the ChemiDoc MP imaging system (Bio-Rad). Settings were as follows: Fluorescein (epi-blue 460–490 nm excitation and 577–613 nm emission filter). After fluorescence imaging, gels were stained in Commassie Brilliant Blue solution for 5 min at room temperature, destained, and then visualized using ChemiDoc MP. Uncropped and unprocessed scans for these and all other blots are provided in the Source Data file.\n",
      "\n",
      "Fluorescence cell imaging of live cells treated with CysOx probes\n",
      "\n",
      "HeLa cells were purchased from ATCC® (#CCL-2). HeLa cells (seeding population: 8.0 × 104 cells/well) were incubated in EMEM (+10% FBS) in 6-well glass bottom plates at 37 °C. At 70–80% confluence, media was aspirated, and cells were washed twice with PBS and serum-free EMEM was added to wells. After incubation for 16 h at 37 °C the media was aspirated, and cells were washed twice with PBS. For GOX treatment, cells were treated with CysOx probe and GOX at the indicated concentrations in PBS containing 0.1% DMSO and glucose (1 mg/mL). At the indicated time points, wells were analyzed using an Olympus FluoView IX81 confocal microscope using FV10-ASW software v3.0. For tBOOH treatment, oxidant was added at the indicated concentrations in EMEM for 10 min at 37 °C followed by the addition of CysOx probe at the indicated concentration for additional 15 min at 37 °C. The media was then aspirated, cells washed twice with PBS, fresh PBS was added, and wells were analyzed as above. Fluorogenic CysOx signal was visualized using the 458 nm laser channel, filtered using a SDM560 dichroic mirror, followed by BA505-605 band pass filter. Five frames were recorded per condition. Images were analyzed using ImageJ software v1.52a, where the pixel intensity of the cellular cytoplasmic regions was measured with at least 10 measurements per condition.\n",
      "\n",
      "Cell viability\n",
      "\n",
      "HeLa cells were incubated in EMEM supplemented with 10% FBS on black 96-well plates with clear bottoms (2 × 104 cells/well) at 37 °C. Once the cells acquired a confluence of 80–90% (after 24-48 h), the media was aspirated, cells were washed twice with PBS and fresh EMEM (lacking FBS) was added. The cells were incubated at 37 °C overnight. The cells were then washed twice with PBS, and PBS (90 µL) was added to each well. The wells were treated with a serial 1:4 dilution of CysOx1 or CyxOx2 (5000, 1250, 313, 78.1, 19.5, 4.88, 1.22, 0.305, 0.076, and 0.019 µM final concentrations) in PBS containing 1% DMSO (total volume/well = 100 µL, 10 wells per condition). One set of ten wells was treated with PBS containing 1% DMSO lacking probe as a viability control for untreated cells. The cells were incubated at 37 °C for 1 h and 100 µL of CellTiter-Glo® reagent (Promega, Cell-titer-Glo® Luminescent Cell Viability Assay, Cat. No. G7570) was added to each well. Plates were mixed for 2 min on an orbital shaker and then incubated at room temperature for 10 min. Luminescence was recorded in a Molecular Devices SpectraMax M5 plate reader using SoftMax Pro software v5.4 with an integration time of 1 s per well.\n",
      "\n",
      "Colocalization analysis of CysOx probe labeling with ER and Mitochondria\n",
      "\n",
      "HeLa cells (8.0 × 104 cells/well) were incubated in EMEM (+10% FBS, supplemented with penicillin/streptomycin) in six-well plates with a glass bottom for imaging (Chemglass, Cat. No. CLS-1812-006). Culture media was aspirated, cells were washed twice with PBS, EMEM lacking FBS was added to the wells, and cells were incubated at 37 °C for 16 h. Culture media was aspirated, and cells were washed twice with PBS. Wells were treated with CysOx1 or CysOx 2 (50 µM final concentration) and an organelle-selective dye (Invitrogen, ER-tracker red, Cat. No. E34250 or Cell Signaling Technology, Mitotracker Deep Red FM, Cat. No. 8778 S) in EMEM with 0.1% DMSO for 1 h at 37 °C. Selected wells were also treated with H 2 O 2 (300 µM final concentration). After incubation, culture media was aspirated, cells were washed twice with PBS, fresh PBS was added, and analyzed using an Olympus FluoView IX81 confocal microscope using FV10-ASW software v3.0.\n",
      "\n",
      "Preparation of lysate from HeLa cells incubated with CysOx probes\n",
      "\n",
      "HeLa cells (2 × 108 cells/well) were scraped into lysis buffer (50 mM HEPES, 100 mM NaCl pH 7.4 containing 0.1% SDS, 1% NP-40, 1X EDTA-free protease inhibitor, and 200 U/mL of catalase) at 4 °C for 20 min. Cell lysates were clarified by centrifugation at 14,000 rpm for 20 min at 4 °C and passed through Zeba spin desalting columns (Pierce, 7 K MWCO) to remove excess probe and protein concentration was determined by BCA assay.\n",
      "\n",
      "Detection of S-sulfenation in EGFR\n",
      "\n",
      "HeLa cells (2 × 108 cells/well) were transfected with wild-type or C797S EGFR pCMV6-XL-448. After 24 h, cells were treated with H 2 O 2 (500 µM) and then CysOx1 or CysOx2 (50 µM) for 1 h at 37 °C. Lysates were prepared as above. EGFR was immunoprecipitated from 500 µg cell lysate (1 mg/ml) with 2 µg mouse anti-EGFR conjugated agarose (Santa Cruz Biotechnology, Cat. No. sc373746 AC, clone A-10) overnight at 4 °C. Resin was collected by centrifugation and washed with cold RIPA buffer (1×) and cold PBS buffer (2×). Bound proteins were eluted by boiling with reducing load dye, resolved by SDS-PAGE, and analyzed for CysOx labeling by in-gel fluorescence as detailed above. For Western blot detection of total EGFR, samples were resolved by SDS-PAGE, transferred to PVDF membrane, and blocked with 3% BSA. Membranes were washed with TBST, and immunoblotting was performed with using anti-EGFR antibodies conjugated to HRP (Santa Cruz Biotechnology, Cat. No. sc-373746 HRP, 1:100). Blots were developed with chemiluminescence (Pierce) and imaged by film. Data were quantified by densitometry using ImageJ v1.52a.\n",
      "\n",
      "Multi-well cell-based screening assay to identify kinase inhibitors that modulate S-sulfenation\n",
      "\n",
      "The ‘Kinase Screening Library’ was purchased from Cayman Chemical (Cat. No. #10505). The library consists of two plates and contains ~160 curated selective and non-selective kinase inhibitors in a 96-well plate format as 10 mM stock solutions in DMSO. This curated library includes inhibitors of a wide range of lipid, receptor and non-receptor tyrosine, serine/threonine, and dual specificity kinases including those belonging to the ROCK, activin-like kinase (ALK), GSK3, PKC, PDGFR, VEGFR, Src, MAPK, CDK, and PI3K families, among many others. It offers coverage of more than 70 distinct kinases and kinase families, as well as numerous additional kinase isoforms and individual kinases within target families. Detailed information on inhibitor selectivity can be found on the manufacturer's website. HeLa cells were incubated in EMEM supplemented with 10% FBS in black 96-well plates with clear bottoms (seeding population: 2 × 104 cells/well) at 37 °C. At 90% confluence (48 h), media was aspirated, cells washed twice with PBS and serum-free EMEM was added to the wells. After incubation for 16 h at 37 °C media was aspirated, cells were washed twice with PBS, and then PBS (90 µL) was added to each well. CysOx2 (50 µM) and kinase inhibitor (10 µM) in PBS containing 1% DMSO was added each well for a total volume 100 µL per well. Cells were then incubated for 1 h at 37 °C. Select control wells did not receive kinase inhibitors and were instead treated with either PBS (negative control) or tBOOH (200 µM, positive control). Following the incubation period, PBS (100 µL) and extracellular fluorescence quencher (20 µL, Solution C of Beta-Lactamase Loading Solutions Kit from Life Technologies, Cat. No. K1048) were added to each well prior to measurement in a Molecular Devices SpectraMax M5 plate reader and analyzed using SoftMax Pro software v5.4. The fluorescence parameters of CysOx2 were selected according to the maxima measured in the fluorescence spectra (excitation/emission: 447/606 nm). Preliminary screening to identify hits was performed in duplicate and compounds exhibiting more than a 2.5-fold increase in fluorescence were subsequently confirmed in quintuplicate according to the same procedure.\n",
      "\n",
      "Measurement of β-catenin phosphorylation and GS activity\n",
      "\n",
      "HeLa cells (2 × 108 cells/well) were treated with vehicle or with 0.1, 1 or 10 µM SB-216763 for 30 min and then with 50 nM calyculin at 37 °C. Lysates were prepared as described above, separated by SDS-PAGE, and transferred to PVDF membrane. Immunoblotting was performed using primary and secondary antibodies at the indicated dilutions in TBST: β-catenin (Cell Signaling, Cat. No. 8480, 1:1000) or phospho-β-catenin-S33/S37/T41 (Cell Signaling, Cat. No. 9561, 1:1000) and goat anti-rabbit IgG-HRP (Calbiochem, Cat. No. 12-348, 1:25,000). Blots were developed with chemiluminescence, imaged by film, and quantified as above. To measure GS activity, serum-starved HeLa cells were treated for 60 min with 10 µM Bio, BIM-IX or SB-216763 or DMSO vehicle. Cells were harvested, non-denaturing lysates prepared, and supernatants assayed for GS activity using the GCS activity assay kit (Cat. No. AK0138-100T-96S, Sunlong) by monitoring the oxidation of NADH oxidation to NAD+ at 340 nm.\n",
      "\n",
      "Live cell HeLa labeling with clickable sulfenic acid probe BTD\n",
      "\n",
      "HeLa cells (2 × 104 cells/well) were incubated for 1 h with different concentrations of SB-216763 (0, 0.01, 0.1, 1, 5, and 10 µM final concentrations) and 1 mM BTD in PBS supplemented with 1% DMSO for selected times. Culture media was aspirated, and cells were washed three times with cold PBS. The supernatant was aspirated, and cells were incubated in 300 µL of lysis buffer (50 mM HEPES, 100 mM NaCl pH 7.4 containing 0.1% SDS, 1% NP-40, 1X EDTA-free protease inhibitor, and 200 U/mL of catalase) at 4 °C for 20 min. Cell lysates were clarified by centrifugation at 16,000 rpm for 20 min at 4 °C. Protein concentration was determined by BCA assay. Lysates were diluted to a concentration of 1 mg/mL (100 µg in 100 µL). Click reactions were performed in each sample: 2 µL of TAMRA-azide (5 mM) was added, followed by addition of a pre-mixed solution of 2 µL CuSO 4 (12.5 mM in ddH 2 O) and 5 µL of BTTP (10 mM in ddH 2 O). Then, 2 µL sodium ascorbate (125 mM in ddH 2 O) was added into each sample, which was incubated at room temperature with gentle shaking for 1 h. Reactions were quenched by addition of EDTA (2 µL, 50 mM in ddH 2 O). Samples were resolved by SDS-PAGE and then analyzed by in-gel fluorescence in the 520 nm channel to observe TAMRA fluorescence in an Azure Sapphire Biomolecular Imager. The gel was then incubated with GelCodeTM Blue Safe Protein Stain (Thermo Scientific) for 30 min and then with water for additional 30 min and imaged.\n",
      "\n",
      "Chemoproteomic S-sulfenome analysis\n",
      "\n",
      "HeLa cells (2 × 108 cells/well) treated with or without GSK3 inhibitor were harvested, lysed in pre-chilled NETN buffer [50 mM HEPES (pH 7.6), 150 mM NaCl, and 1% IGEPAL] supplemented with protease and phosphatase inhibitors (Thermo Scientific, Cat. No. A32961) containing 200 U/mL catalase (Sigma-Aldrich), and then incubated with 5 mM of BTD at 37 °C for 2 h with rotation and light protection. The resulting probe-labeled samples were incubated with 40 mM iodoacetamide at 37 °C for 1 h with light protection. To remove excess small molecules, proteins were precipitated by methanol-chloroform (aqueous phase/methanol/chloroform, 4:4:1 (v/v/v)). The precipitated proteins were resuspended with 50 mM ammonium bicarbonate containing 0.2 M urea and digested with sequencing grade trypsin (Promega) at a 1:50 (enzyme/substrate) ratio overnight at 37 °C. The tryptic digests were desalted, evaporated to dryness, resuspended in a water solution containing 30% acetonitrile (ACN) and subjected to CuAAC ligation24. After 2 h incubation at room temperature, the light and heavy reaction mixtures were then combined, purified by strong cation exchange (SCX), and then subjected to the enrichment with streptavidin beads for 2 h at room temperature. Streptavidin beads were washed with 50 mM NaAc (pH4.5), 50 mM NaAc containing 2 M NaCl (pH4.5), and deionized water twice each with vortexing and/or rotation to remove non-specific binding substances, then resuspended in 25 mM ammonium bicarbonate, photo-released under 365 nm UV light for 2 h at room temperature with magnetic stirring. The supernatant was collected, dried under vacuum, and stored at −20 °C until further analysis. LC-MS/MS analyses were performed on a Q Exactive plus instrument (Thermo Fisher Scientific). Peptide samples were reconstituted in 0.1% formic acid and pressure-loaded onto a 2-cm microcapillary precolumn packed with C18 (3-μm, 120 Å, SunChrom, USA) operated with an Easy-nLC1000 system (Thermo Fisher Scientific). The precolumn was connected to a 12-cm 150-μm-inner diameter microcapillary analytical column packed with C18 (1.9-μm, 120 Å, Dr. Maisch GebH, Germany) and equipped with a homemade electrospray emitter tip. The spray voltage was set to 2.0 kV and the heated capillary temperature to 320 °C. The LC gradient consisted of 0 min, 7% B; 14 min, 10% B; 51 min, 20% B; 68 min, 30% B; 69–75 min, 95% B (A = water, 0.1% formic acid; B = MeCN, 0.1% formic acid) at a flow rate of 600 nL/min. MS1 spectra were recorded with a resolution of 70,000, an AGC target of 3e6, a max injection time of 20 ms, and a mass range from m/z 300 to 1400. HCD MS/MS spectra were acquired with a resolution of 17,500, an AGC target of 1e6, a max injection time of 60 ms, a 1.6 m/z isolation window and normalized collision energy of 30. Peptide m/z that triggered MS/MS scans were dynamically excluded from further MS/MS scans for 18 s. Raw data files were searched against Homo sapiens Uniprot canonical database using pFind studio software v3.1.249. Precursor ion mass and fragmentation tolerance were set as 10 ppm and 20 ppm, respectively. The maximum number of modifications and missed cleavages allowed per peptide were both set as three. For all analyses, mass shifts of +5.9949 Da (methionine oxidation) and +57.0214 Da (iodoacetamide alkylation) were searched as variable modifications. For site-specific mapping of probe-modified sulfenic acid sites, mass shift of +418.131 (C 19 H 22 N 4 O 5 S) for BTD was searched as variable modification. A differential modification of 6.0201 Da on probe-derived modification was used for stable-isotopic quantification. The FDRs were estimated by the program from the number and quality of spectral matches to the decoy database. The FDRs at spectrum, peptide, and protein level were <1%. Note that identifications based on single PSM (peptide-spectrum match) was allowed, which were manually inspected50. For those identified peptides containing multiple cysteine residues, at least three modification-specific fragment ions were required for unambiguous site localization. Quantitative analyses from two biological replicates were performed using pQuant software v1.051.\n",
      "\n",
      "Statistics and reproducibility\n",
      "\n",
      "Measurements were taken from distinct samples. Data are reported as mean ± SEM unless otherwise indicated in the figure legend. Statistical analyses were performed using one- or two-way ANOVA with Tukey’s honest significant difference test correction for multiple comparisons using GraphPad Prism v9.4. For all analyses, p < 0.05 was considered significant.\n",
      "\n",
      "Reporting summary\n",
      "\n",
      "Further information on research design is available in the Nature Research Reporting Summary linked to this article.\n",
      "n\n",
      "Article's Summary:\n",
      "C64,82 S Gpx3-SH was exchanged into labeling buffer (50 mM HEPES, 100 mM NaCl, pH = 7.4) using P-30 spin columns (Bio-Rad).\n",
      "After incubation for 16 h at 37 °C the media was aspirated, and cells were washed twice with PBS.\n",
      "The media was then aspirated, cells washed twice with PBS, fresh PBS was added, and wells were analyzed as above.\n",
      "The cells were then washed twice with PBS, and PBS (90 µL) was added to each well.\n",
      "After 24 h, cells were treated with H 2 O 2 (500 µM) and then CysOx1 or CysOx2 (50 µM) for 1 h at 37 °C.\n",
      "n\n"
     ]
    }
   ],
   "source": [
    "url2 =\"https://www.nature.com/articles/s41467-022-33124-z\"\n",
    "toi2_article = Article(url2, language=\"en\")\n",
    "toi2_article.download()\n",
    "toi2_article.parse()\n",
    "toi2_article.nlp()\n",
    "print(\"Article's Title:\")\n",
    "print(toi2_article.title)\n",
    "print(\"n\")\n",
    "print(\"Article's Text:\")\n",
    "print(toi2_article.text)\n",
    "print(\"n\")\n",
    "print(\"Article's Summary:\")\n",
    "print(toi2_article.summary)\n",
    "print(\"n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Article's Title:\n",
      "Structural materials with afterglow room temperature phosphorescence activated by lignin oxidation\n",
      "n\n",
      "Article's Text:\n",
      "RTP emission of OL\n",
      "\n",
      "The ultraviolet-visible (UV-Vis) spectra of OL (prepared from lignosulfonates, see the details in the methods) exhibits optical absorbance predominantly in the UV region, over the wavelength range from 200 nm to 400 nm (Supplementary Fig. 1). OL exhibited a decreased absorbance, compared to the raw lignin (Supplementary Fig. 2). Raw lignin and OL also exhibited obvious differences in the FT-IR spectra (Supplementary Fig. 3). The signal intensity of -C = O at 1704 cm−1 and -O-H at 3300 cm−1 increased after lignin oxidation. A decrease in the molecular weight of lignin after oxidation was observed using gel permeation chromatography (GPC) (Supplementary Fig. 4). The weight-average molecular weight of lignin decreased from 11541 to 1588 after oxidation. Thermogravimetric analysis (TGA) and derivative thermogravimetry (DTG) were also investigated before and after lignin oxidation (Supplementary Fig. 5). These results indicated that lignin degraded into small fragments after oxidation. Upon UV irradiation, OL exhibited strong fluorescence emission, centered at ~464 nm (Fig. 2a). Surprisingly, afterglow RTP emission was also observed, with phosphorescence centered at ~525 nm. Time-resolved spectroscopy indicated that the OL displays a long-lasting and stable afterglow emission and retained weak phosphorescence emission at the same emission wavelength for as long as 800 ms (Fig. 2b). The lifetime of OL was ~408 ms (Supplementary Fig. 6). Moreover, excitation-independent RTP emission of OL was observed. The RTP emission wavelength red shifted from 460 nm to 550 nm when the excitation wavelength changed from 275 nm to 400 nm, indicating multiple chromophores in OL (Fig. 2c). Additionally, to illustrate the generality of our strategy, another technical lignin, alkali lignin, was also treated using the same method to obtain OL. As expected, the as-prepared OL from alkali lignin produced efficient afterglow RTP emission with a lifetime of ~215 ms (Supplementary Fig. 7). The afterglow RTP was found to be humidity-sensitive, with reduction of the phosphorescence lifetime when the humidity was gradually increased. In particular, the phosphorescence lifetime dropped rapidly from ~408 ms to ~102 ms when the humidity increased from 10% to 70% (Fig. 2d). To understand the effect of humidity on the lifetime, OL was exposed to different levels of humidity. The results indicated that the lifetime of OL was completely quenched when the humidity reached 100% (Supplementary Fig. 8). More interestingly, this quenched lifetime was recovered again after drying the OL. The “humidity-drying” sensitive RTP lifetime was stable and it could be recycled 7 times (Supplementary Fig. 9). Notably, OL exhibited a decreased lifetime upon exposure to increased temperature (from 100 −175 oC) for an extended time (~90 min) (Supplementary Fig. 10). It was possibly attributed to decomposition of lignin, as determined by TGA and DTG analysis. Additionally, OL displayed mechanoresponsive RTP. The afterglow RTP lifetime decreased from ~408 ms to ~248 ms when the external pressure increased from 0 MPa to 60 MPa (Supplementary Fig. 11). Additionally, the afterglow emission of OL was stable in organic solvents. The lifetimes were 489.34, 516.55 and 519.84 ms in DCM, ETOH and CH 3 CN, respectively (Supplementary Fig. 12).\n",
      "\n",
      "Fig. 2: Afterglow RTP emission of OL. a Fluorescence (PL.) and RTP (Phos.) emission of OL, excitation wavelength = 365 nm, inset: Images of bright field, fluorescent and RTP emission of OL upon UV irradiation (365 nm). b Time-resolved RTP emission of OL, excitation wavelength = 365 nm. c Excitation-dependent RTP emission of OL; d RTP lifetime of OL upon exposure to humidity and drying. Full size image\n",
      "\n",
      "Mechanism investigation\n",
      "\n",
      "To gain a deeper insight into the mechanism of afterglow RTP emission of OL, a set of experiments was conducted. The 2D heteronuclear singular quantum correlation nuclear magnetic resonance (2D HSQC NMR) indicated that lignin mainly consists of G units and S units in the aromatic region and the ratio of G and S units was 10:90 (Fig. 3a). Lignin also displayed abundant signals between 3.0 ppm and 6.0 ppm, which was attributed to the C-O-C and C-C linkers (Supplementary Fig. 13). After oxidation, most of the signals in the aromatic region (6.2–6.8 ppm, 103–115 ppm) disappeared. In response, the signals of G acids and S acids were observed at 6.6 ppm and 134 ppm, which were exactly the oxidated products from G and S units, respectively. While, OL displayed new signals between 3.0 ppm and 6.0 ppm (Supplementary Fig. 13), probably from fatty acids. To further verify the structural evolution of lignin, high-resolution MS (HRMS) analysis was conducted on an OL sample. As expected, G/S acids were observed in the HRMS spectra (Supplementary Fig. 14). Moreover, fatty acids including sulfonic fatty acids and fatty carboxylic acids were detected, confirming that the 2D HSQC NMR signals between 3.0 and 6.0 ppm were due to fatty acids (Supplementary Fig. 14). Pyrolysis gas chromatography-mass spectrometry (Py-GC-MS) analysis was conducted to further understand the structural change. The results indicated that lignin had degraded into smaller fragments after oxidation (Supplementary Fig. 15 vs Supplementary Fig. 16). According to literature34,35,36, and as-obtained HRMS, Py-GC-MS, and GPC results, the proposed pathways for the oxidative depolymerization of lignin into aromatic acids and fatty acids are shown in Supplementary Fig. 17, which went through oxidative C α -C β cleavage and oxidative ring-open reactions, respectively. All the above results indicate that OL contains both aromatic acids and fatty acids. Theoretical simulation further indicated that the aromatic acids (taking vanillic acid as a model compound) could be “locked” by fatty acids (taking succinic acid as a model compound) by hydrogen bonding in OL (Fig. 3b). The simulation results indicated that carbonyl moieties of the aromatic acids formed intermolecular hydrogen bonds, restricting vibration, which enhances spin-orbit coupling and encourages phosphorescence (Fig. 3c)27. Additionally, the results indicated that the average center-center distance between adjacent molecules was ca. ~0.37 nm, facilitating the formation of strong π - π interactions between the aromatic acids (Fig. 3d). The relative orientation of two neighboring model compounds (θ) was populated over a range from ~60o to 100o with an optimal angle of ~95o, indicating H-type dimers formed between them (Fig. 3e)27,37. Such H-type dimers can stabilize the lowest excited triplet states, and prolong the RTP lifetime. Moreover, the calculation result indicated that the spin-orbit coupling ξ (S 1 , T n ) of the H-type dimers is larger than that of monomer molecule, facilitating phosphorescence (Fig. 3f). Additionally, urea, as a reagent for breaking hydrogen bonding and reducing the interaction between aromatic acids and fatty acids, was added to powdered OL38. Immediately, the lifetime of OL decreased to ~212 ms when the urea fraction increased to 28% (Supplementary Fig. 18). To demonstrate how the urea affected the OL, theoretical calculations were performed. The interaction force between model aromatic acids and fatty acids was 79.9 kJ/mol and the intermolecular interaction between fatty acids was 78.7 kJ/mol. However, the interaction between urea and fatty acids was 83.7 kJ/mol, which was higher than the interactions with OL and as such can break the hydrogen bonding between aromatic acids and fatty acids (Supplementary Fig. 19). Moreover, the hydrogen bond formed between fatty acids and urea was confirmed by a downfield shift of the carboxyl and amino protons in the 1H NMR (Supplementary Fig. 19). According to the literature39, hydrogen bonds can induce a decrease of electron density near the hydroxyl proton and deshield the nuclei, resulting in a downfield shift of the hydrogen-bonded protons. The above 1H NMR results provide direct experimental evidence for the formation of hydrogen bonds between succinic acid and urea. Additionally, humidity-responsive lifetimes could be attributed to decreased hydrogen bonding between the fatty acids and aromatic acids. This was confirmed by theoretical calculations. Water exhibits stronger binding energy with fatty acids (82.6 kJ/mol) than aromatic acid. As a result, the addition of water can break the hydrogen bonding between the fatty acids and aromatic acids, facilitating the free rotation of the carbonyl moieties and decreasing SOC and ISC. As such, the lifetime decreased upon exposure to high humidity. (Supplementary Fig. 19). Based on all the above results, a plausible mechanism was proposed. Firstly, the oxidation of lignin by H 2 O 2 produced aromatic acids and fatty acids. After that, the aromatic acids became “locked” by fatty acid hydrogen bonds. Finally, the “locking” restricted the vibration of the carbonyl moieties and encouraged the formation of H-type dimers, which promoted spin-orbit coupling and afterglow emission.\n",
      "\n",
      "Fig. 3: Mechanism of afterglow RTP of OL. a 2D HSQC NMR of lignin (left) and OL (right) in the aromatic region. b and c, Simulation of molecular interactions between model aromatic acids and fatty acids in OL. d Center-center distance between adjacent aromatic molecules in OL. e, Relative orientation of two neighboring aromatic model compounds in OL. f Spin orbit coupling ξ (S 1 , T n , n = 1, 2, 3…7) of vanillic acid dimers and vanillic acid monomer. In the Figure b, c, d and e, Red dot: oxygen atom, Black dot: carbon atom, White dot: hydrogen atom; Black line: chemical bond (-C-C/-C = C-); Dot line: hydrogen bond. Full size image\n",
      "\n",
      "Activating RTP of natural wood\n",
      "\n",
      "Inspired by this discovery, naturally-occurring lignin in wood cell walls was oxidized in situ to generate OL to generate RTP wood. In addition, a smart manufacturing line for RTP wood was constructed. In this line, an automatic manipulator was used to control the immersion of the wood in NaOH and H 2 O 2 solution (Supplementary Fig. 20). Specifically, wood was pretreated by NaOH solution to make the lignin in the cell walls more accessible40. After that, H 2 O 2 solution facilitated in situ oxidation. Thanks to the smart manufacturing line, RTP wood was prepared efficiently (Supplementary Movie 1). UV-Vis spectra indicated that the absorbance intensity at 250-350 nm, was due to the absorbance of lignin41, which decreased significantly after in situ oxidation of the natural wood (Fig. 4a). The as-prepared RTP wood from Basswood after drying exhibited long afterglow RTP emission centered at 510 nm with a lifetime of ~431 ms (Figs. 4b-d). As a control, untreated Basswood displayed phosphorescence emission with a lifetime of 28.25 ms, ~15 times lower than that for RTP wood (Fig. 4e and Supplementary Fig. 21)17. Notably, the wood became more hydrophilic after oxidation, as demonstrated by the decrease in contact angle (from 108 to 55 degree) (Supplementary Fig. 22). Such changes led to a sensitive response of RTP wood to water. Immersing the wood in water immediately quenched its emission (Supplementary Fig. 23). However, the problem can be solved by coating RTP wood with hydrophobic wax. Such coated RTP wood exhibited afterglow emission when it was immersed in water (Supplementary Fig. 24). Moreover, the method for producing RTP wood was widely applicable. For example, different types of wood including peach wood, rubber wood, and Schima superba were efficiently converted to RTP woods using the in situ oxidation method (Fig. 4f-i and Supplementary Fig. 25–27). The lifetimes of RTP wood made from peach wood, rubber wood, and Schima superba increased from 19.53, 70.35 and 60.95 to 284.00, 197.31 and 224.91 ms, respectively (Supplementary Fig. 28). Interestingly, a patterned RTP wood could be achieved by the selective oxidation of the wood surface. As such N shape and 2-D code were patterned on natural wood. After switching off the UV light source, afterglow RTP images could be clearly observed (Fig. 4j, k). Moreover, the 2-D code could be recognized using a smartphone (Fig. 4k and Supplementary movie 2). As a practical demonstration, a series of model afterglow furniture were constructed using RTP wood (Fig. 4l). Upon light irradiation, the furniture made from RTP wood exhibited nice fluorescence. In addition, nice afterglow RTP was observed after switching off the light source. Considering the effect and importance of sustainable indoor lighting materials for the physical and psychological well-being of building occupants, our afterglow furniture has great potential for house decorations.\n",
      "n\n",
      "Article's Summary:\n",
      "Moreover, fatty acids including sulfonic fatty acids and fatty carboxylic acids were detected, confirming that the 2D HSQC NMR signals between 3.0 and 6.0 ppm were due to fatty acids (Supplementary Fig.\n",
      "The interaction force between model aromatic acids and fatty acids was 79.9 kJ/mol and the intermolecular interaction between fatty acids was 78.7 kJ/mol.\n",
      "Additionally, humidity-responsive lifetimes could be attributed to decreased hydrogen bonding between the fatty acids and aromatic acids.\n",
      "Firstly, the oxidation of lignin by H 2 O 2 produced aromatic acids and fatty acids.\n",
      "As a practical demonstration, a series of model afterglow furniture were constructed using RTP wood (Fig.\n",
      "n\n"
     ]
    }
   ],
   "source": [
    "url3 =\"https://www.nature.com/articles/s41467-022-33273-1\"\n",
    "toi3_article = Article(url3, language=\"en\")\n",
    "toi3_article.download()\n",
    "toi3_article.parse()\n",
    "toi3_article.nlp()\n",
    "print(\"Article's Title:\")\n",
    "print(toi3_article.title)\n",
    "print(\"n\")\n",
    "print(\"Article's Text:\")\n",
    "print(toi3_article.text)\n",
    "print(\"n\")\n",
    "print(\"Article's Summary:\")\n",
    "print(toi3_article.summary)\n",
    "print(\"n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Article's Title:\n",
      "Adar-mediated A-to-I editing is required for embryonic patterning and innate immune response regulation in zebrafish\n",
      "n\n",
      "Article's Text:\n",
      "Adar and Adarb1b are expressed during zebrafish development\n",
      "\n",
      "To determine whether A-to-I editing activity exists during embryonic development, we revisited our transcriptome data36 to check whether the enzymes responsible for A-to-I editing were expressed. In agreement with other recently published data23, we detected transcripts of at least two deaminase paralogs, adar and adarb1b, from egg to 5.3 hpf (Fig. 1a). Transcripts of these two paralogs were present both maternally as well as zygotically, with adar being more abundant (more than 4-fold compared to adarb1b at each developmental stage). Moreover, we found that transcripts of both paralogs were consistently associated with polysome, starting from the egg stage up to 5.3 hpf (Fig. 1b). This finding suggests that these transcripts are constantly undergoing translation at developmental stages preceding and after the activation of zygotic genome36. The observation that both paralogs were already expressed, and their transcripts associated with polysomes at egg stage suggests that RNA editing events occur prior to fertilization and may be crucial for early development. Interestingly, a substantial increase in adar expression occurs after the mid-blastula transition (MBT), suggesting that the role of this gene extends beyond the period of transcriptional silence in early embryogenesis (Fig. 1a). At larval stage, the expression of both gene paralogs was not spatially restricted although more abundant in the nervous system of the developing embryo as shown by whole mount in situ hybridization of adar (Fig. 1c, d) and adarb1b (Fig. 1e, f) in 24 hpf zebrafish embryos.\n",
      "\n",
      "Fig. 1: Spatiotemporal expression pattern of Adar family genes in zebrafish. a Expression of zebrafish adar family based on transcriptome profiling of developing embryos (RNA-seq)53. Expression levels are plotted for developmental stages: egg, 1 cell, 16 cells, 128 cells, 3.5 hpf, and 5.3 hpf. Levels of a total RNA and b polysome-associated RNA of four zebrafish adar paralogs are given. Whole mount in situ hybridization shows expression pattern of adar (c, d) and adarb1b (e, f) in 24 hpf zebrafish embryos. Each experiment was performed three times on embryos from three independent mating pairs with similar results. Source data are provided as a Source data file. Full size image\n",
      "\n",
      "Adar editing activity is required for embryonic patterning\n",
      "\n",
      "To investigate whether Adar plays a biological role in early zebrafish development, we used morpholino oligonucleotides (MO) to knockdown Adar, the highest expressed among all zebrafish ADAR paralogs (Fig. 1a). Adar MO-injected embryos developed a range of phenotypes which is initially evident at gastrulation and subsequently observed to affect posterior body axis by 24 hpf (Fig. 2a–c, h). The most severe phenotypes manifested in a lack of almost all posterior structures and crooked body axis (Fig. 2c). In addition, the notochord in morphant embryos was disorganized, with unevenly shaped and distributed vacuoles instead of the neat “stack of coins” arrangement in wild-type (Fig. 2b, c). This abnormal MO phenotype was dose dependent and could be rescued with the wild-type adar mRNA, in which an increased percentage of embryos appeared normal or exhibited mild phenotype with proper body axis organization and tail length (Fig. 2d, h; Supplementary Fig. 1). On the other hand, knockdown with up to 2 ng of MO against Adarb1b, which was expressed at a lower level in the early embryo, did not result in any observable phenotype (Supplementary Fig. 1). To verify whether the biological role of Adar depends on its RNA editing activity, we generated adar mRNA E1030A designed after a similar construct in mammals37, in which the deaminase domain was mutated (Supplementary Fig. 1). The E1030A mRNA was unable to rescue MO-induced phenotype in developing embryos, resulting in comparable number of abnormally developed embryos with severe posterior axis defects to that of MO-injected larvae (Fig. 2e, h). This suggests that a functional deaminase domain, catalyzing the A-to-I editing in dsRNA, is essential for early embryonic development. These results strongly suggest that the A-to-I editing activity of Adar is necessary for the specification of early embryonic axes.\n",
      "\n",
      "Fig. 2: Phenotypic defects at 24 hpf caused by adar knockdown and overexpression. a Wild-type (b, c) Adar MO-injected embryos develop abnormal phenotype in the posterior part with disturbed body axis, shortened tail and crooked, disorganized notochord. d MO phenotype can be fully rescued with wild-type mRNA injection. e Mutant adar mRNA E1030A with inactivated editing domain could not rescue the malformed phenotype. f, g Phenotype defects caused by adar mRNA overexpression. The anterior defects, including cyclopia and head malformations are mRNA dose dependent. Inset marked by red boxes denotes overlaid identical image taken at different focal plane. Observation was performed on embryos from four independent adult pairs with similar results. h, i Injection statistics of Adar MO, rescue, and mRNA overexpression. Source data are provided in the Source data file. Full size image\n",
      "\n",
      "Whereas Adar MO affected the development of the posterior part of the body, adar mRNA overexpression caused significant abnormalities of the anterior part, which include anomalous eye development, most often manifested in cyclopia, deformed cranium, and reduced or absent brain compartment, particularly at the anteriormost region (Fig. 2f, g, i). These defects were most prominently observed with 50 pg injection of mRNA. The effect of adar mRNA overexpression was dose dependent with a 64% mortality for the 100 pg of injected mRNA, 9% for 50 pg and 7% for 25 pg (compared to wt which had 5% mortality). Overexpression of the E1030A mRNA did not lead to an abnormal phenotype, further confirming that the function of Adar is deaminase-dependent (Supplementary Fig. 1B). Collectively, our results show that Adar plays a key role in the earliest steps of embryonic patterning.\n",
      "\n",
      "In order to characterize the observed morphological defects of Adar loss- and gain- of function in more detail, we assessed the expression of several marker genes indicating various embryonic structures. The most prominent phenotype of Adar disruption is that of the development of structures along the anteroposterior axis. In order to characterize the effect of Adar loss- or gain-of function on the development of anterior embryonic structures, we utilized the expression of pax6 and tbx2b to collectively indicate the forebrain, midbrain, and hindbrain, anterior spinal cord, dorsal retina, and otic vesicle (Fig. 3a, d, g, j)38,39. In adar morphants where morphological defects were predominantly observed in the posterior region, these anterior structures were preserved and appeared similar to wild-type in terms of their size and organization (Fig. 3b, e, h, k). On the contrary, anteriormost brain regions and retina were indistinguishable in embryos overexpressing adar (Fig. 3c, f, i, l). In particular, pax6 expression shows that the forebrain expression domain were unrecognizable in adar-overexpressing embryos (Fig. 3c, f). Moreover, tbx2b expression domains of diencephalon, left and right retina, and the epiphysis appeared as a fused region, whereas the more posterior expression domains of the trigeminal ganglion and otic vesicles appeared less affected although hypomorphic compared to control (Fig. 3i, l). While the eye field is derived from the anterior neural plate and therefore arose as a consequence of anteroposterior patterning40, its failure to split in the midline (cyclopia) is a hallmark of a defect in the convergent-extension movement during gastrulation which is dependent on the establishment of the dorsoventral axis41,42. The observed phenotype of adar overexpression therefore suggests that proper establishment of these two body axes was affected.\n",
      "\n",
      "Fig. 3: Embryonic patterning defects due to Adar loss- or gain-of-function. Whole mount in situ hybridization detection of marker genes in Adar morpholino knockdown and adar mRNA overexpression. a–f pax6 expression demarcates anterior brain structures including dorsal retina (R), forebrain (FB) as well as hindbrain (HB) and anterior spinal cord. g–l tbx2b expression marking diencephalon (DE), R, epiphysis (EP), trigeminal ganglion (TG), and otic vesicle (OV). m–o expression of tbxta marks the notochord. p–r shhb expression indicates floor plate (FP) and midbrain-hindbrain boundary (MHB). The experiment was performed three times on embryos from three different mating pairs with similar results. Full size image\n",
      "\n",
      "To examine the defects in posterior structures, we used as reporter genes tbxta and shhb which mark the notochord and floor plate neurons of the spinal cord, respectively (Fig. 3m, p). Similar to the anterior markers, shha and tbxta were expressed in the case of both adar loss- or gain-of-function, suggesting that the neural and midline cell identities were preserved (Fig. 3n, o, q). However, the body axis including the notochord and spinal cord were crooked in adar morphants as well as in adar overexpressing embryos albeit to a lesser extent in the latter (Fig. 3n, o). Importantly, loss- or gain-of-function of adar do not appear to affect the specification of cell identity as evidenced by the preserved expression of pax6, tbx2b, shha, and tbxta. Rather, a more pronounced effect is seen on the gross anteroposterior and dorsoventral patterning of the embryo. Collectively, these observations suggest that Adar-mediated A-to-I editing may be involved in the patterning along the two early embryonic axes of the zebrafish.\n",
      "\n",
      "Prevalent A-to-I editing in maternal and early zygotic transcripts\n",
      "\n",
      "The effects of Adar disruption on embryonic patterning led us to ask to what extent A-to-I editing occurs in early embryogenesis, and whether transcripts of genes involved in this process were specifically affected. To profile global A-to-I RNA editing in zebrafish we sequenced a trio of wild-type sample of both parental genomes and the transcriptome of their offspring at three developmental stages: 1.5 hpf (16 cells, pre-MBT), 3.5 hpf (high, MBT) and 5.3 hpf (50% epiboly, post-MBT) (Fig. 4a). Comparison of zygotic transcriptomes with the genomic sequence of parents allowed us to pinpoint RNA editing events by identifying mismatches between the genomic and transcriptomic reads. As Inosine (I) is structurally similar to a G due to the presence of the 6-oxo group, reverse transcriptase incorporates a C in the corresponding position during RNA-seq library synthesis; thus, in the original transcript strand, a G is inferred. Therefore an A-to-I editing event can be identified as an A–G mismatch between the parental genome and the corresponding embryonic transcript. Since the RNA-seq was unstranded, the same applies as well to a T–C mismatch. Strikingly, our analyses revealed a disproportionate enrichment of A–G and T–C mismatches compared to other possible base mismatches which could occur stochastically at all three developmental stages (Fig. 4b). This strongly suggests the presence of A-to-I RNA editing in maternally deposited as well as in zygotic transcripts. Altogether, we identified 44,007 RNA editing sites: 11,374 of which were shared by all three samples, 6352 between 1.5 hpf and 3.5 hpf, 1393 between 1.5 hpf and 5.3 hpf, and 1,564 between 3.5 hpf and 5.3 hpf. Apart from this, 6117, 5686 and 11,521 were specific to 1.5 hpf, 3.5 hpf and 5.3 hpf, respectively (Fig. 4c, Supplementary Data 1). These stage-specific editing events indicate differential patterns of RNA editing throughout early embryogenesis and suggest that this process may play a role in embryogenesis. Interestingly, merely 2% of RNA editing sites occurred within coding sequences (CDS), while the majority (38% in 1.5 hpf and 3.5 hpf and 27% in 5.3 hpf) occurred in 3’-UTR regions (Fig. 4d, Supplementary Data 1). A large fraction of editing was assigned as ‘genic_other’ due to overlap between intron/exon/UTRs from multiple transcripts. Finally, ~17% of editing events in each stage were detected in ‘intergenic sequence’, but this may be due to unannotated transcripts. The low abundance of A-to-I editing present in the coding regions (CDS), none of which resulting in missense mutations, ruled out the possibility that editing functions through expanding the coding repertoire of expressed transcripts by altering amino acid composition. We also observed plenty of editing occurring in DNA repeat regions with a notable increase in editing frequency in retrotransposons (LTR, LINE, and SINE) in 5.3 hpf compared to earlier developmental stages (Fig. 4e, Supplementary Data 2).\n",
      "\n",
      "Fig. 4: A-to-I RNA editing sites in early embryonic development. a Schematics of A-to-I RNA editing discovery through sequencing of a parent-offspring trio. The genome sequence of the parents are used as a reference set to distinguish between polymorphisms and editing. b Mismatches between RNA and DNA sequencing data. As RNA libraries were not strand selective, mismatches were read as their complement (i.e., T->C instead of A->G, or C->T as G->A) in roughly half of all cases. c Overlap of editing sites at different time points. The 1.5 and 3.5 hpf samples were more similar to each other than to the 5.3 hpf sample, probably because of replacement of maternal by zygotic transcripts at the MZT. d Association of editing sites with genomic features. A large fraction of RNA editing is classified as “genic_other” due to overlap between introns/exons/UTRs from multiple transcripts. e Number of editing sites in transcripts stemming from different classes of repeat elements. f Number of editing events in individual reads encompassing 10 potential editing sites. The majority of individual reads contained 1–3 RNA editing sites, and never more than five editing sites. Source data: Supplementary Data 1–3, Source data file. Full size image\n",
      "\n",
      "To assess the penetrance of A-to-I editing on individual RNA molecules, we calculated the occurrence of editing events in RNA-seq reads. Based on alignment of 75 bp reads, we observed a substantial number of reads with no editing at all for a given region (Fig. 4f). Interestingly, the number of RNA editing sites in a single read never exceeds 50% of the total number of editing events identified in the given transcript region (Fig. 4f, Supplementary Data 3). This suggests that A-to-I editing rarely occurs at all possible editing sites in a given RNA molecule, although sequencing with much longer reads is required to precisely determine the true penetrance. We then ask whether RNA editing events were associated with translation efficiency and/or RNA stability. Using our previously published polysome profiling dataset to determine translation rate36, we found that transcripts undergoing more active translation tend to be less edited than those not associated with polysome (Supplementary Fig. 2). Interestingly, this is true only for editing events detected at post-MBT stages (3.5 hpf, r = −0.053, p = 3.79e−11;and 5.3 hpf, r = −0.071, p = 2.36e−22), but not for pre-MBT (1.5 hpf, r = 0.002, p = 0.78). With regards to RNA stability, no obvious difference was observed in expression levels between low (1–9 sites) and high (more than 10 sites) edited transcripts when 1.5 hpf and 3.5 hpf stages were compared. However, highly edited transcripts undergo much lower expression change between 1.5 hpf and 5.3 hpf (Supplementary Fig. 3).\n",
      "\n",
      "We sought to identify which genes were subject to A-to-I editing at each developmental stage. We identified 639, 634, and 562 genes having at least two editing sites at any position within their transcript at 1.5 hpf, 3.5 hpf, and 5.3 hpf respectively (Supplementary Data 4). Among these, 9, 9, and 10 genes had editing sites occurring in both 3’UTR and coding sequence at each respective stage. Interestingly, although no developmentally relevant GO terms were found to be significantly enriched among edited genes (Supplementary Data 5), we found several key genes known to be implicated in anteroposterior and dorsoventral patterning (Fig. 5). Of note, members of the Wnt signaling pathway were found to be edited. Several Frizzled receptors of Wnt signaling, fzd3b, fzd5, and fzd8b were edited at both maternal stages, while fzd7b were edited at 5.3 hpf. The Wnt downstream effector tcf7l1b was found to be consistently edited at all three stages. We also found two members of the FGF signaling pathway, fgfr1a and extl3, consistently edited at all three stages observed. Interestingly, fgfr1a is one of the most highly edited transcripts, containing 99 edited sites in both its coding sequence and 3’-UTR. Mutation of FGFR1 in humans is associated with holoprosencephaly43 which is reminiscent of the observed adar overexpression phenotype. Other genes which were consistently edited throughout the three stages include furina which plays a role in craniofacial development44 and two genes involved in gastrulation, dusp4 and ezrb45,46. That transcripts of these genes were found to be edited throughout maternal and early zygotic stages suggests the role of A-to-I editing in regulating multiple aspects of anteroposterior and dorsoventral patterning.\n",
      "\n",
      "Fig. 5: Selected transcripts of developmental signaling pathway genes implicated in dorsoventral and/or anteroposterior patterning containing two or more RNA editing sites detected at their 3′-UTR region. Gene names in red denotes those that are commonly edited at all three stages. Source data: Supplementary Data 4. Full size image\n",
      "\n",
      "Adar loss of function affects dorsoventral patterning\n",
      "\n",
      "To uncover the role of Adar at the molecular level, we performed transcriptome sequencing of wild-type, Adar MO knockdown (KD) and Adar overexpressing (OE) embryos at 128 cells (before ZGA), 5.3 hpf (after ZGA), and 12 hpf (when embryonic patterning is established). We hypothesized that alterations in the level of adar transcripts may evoke consequential differences in mRNA editing and/or changes in the expression of potential target genes. To this end, we performed comparative transcriptome analysis of wt vs. OE and wt vs. KD in each developmental stage. Surprisingly, no substantial changes in global A-to-I editing levels were found in KD and OE conditions at 128-cell and 5.3 hpf stages (Supplementary Fig. 4A). Moreover, during these two stages, samples clustered according to developmental stage (Supplementary Fig. 4B) and no substantial changes were observed in global gene expression profile (p < 0.05; Supplementary Fig. 4C). Only 539 sites have a slightly elevated editing level in OE than in KD in both developmental stages (Supplementary Data 6).\n",
      "\n",
      "In contrast, at 12 hpf, a more noticeable change in both global editing pattern and gene expression profile was observed between control and Adar gain- or loss-of-function at 12 hpf (Fig. 6a; Supplementary Fig. 5). Unlike at earlier stages, samples clustered according to conditions rather than developmental stage (Fig. 6b). Interestingly, in all three replicates, Adar KD caused a modest but more noticeable change in editing frequency compared to the earlier stages (Supplementary Fig. 5). Moreover, at 12 hpf, we observed 827 and 5054 genes differentially expressed in Adar KD and Adar OE respectively, compared to control (p < 0.05; Supplementary Data 7). GO analysis (Supplementary Data 8) revealed that Adar KD generally caused the upregulation of genes regulating epiboly (nanog, cacnb4b), gastrulation (bcl2l10, mylipa), and ectoderm development (pou5f3, cdh1), while concurrently causing downregulation of those implicated in convergent-extension (wnt11f2, creb1a, ppp1cb) and mesoderm development (tbx16, hes6, her7, mcdh2, myf5, msgn1). On the other hand, Adar OE resulted in the upregulation of genes involved in the development of mesodermal structures (myf6, bves, tcf21, tbx5a, apln, mef2aa), while downregulating genes regulating epiboly (chuk, epcam, mapkapk2a, cldne), dorsoventral pattern formation (sox11b, dusp6, ved, vox, bambia, acvr1ba, ctnnb2), and brain development. Interestingly, out of the 383 genes common between the two conditions, 283 commonly downregulated genes in both Adar KD and OE included genes known for their role in anterior-posterior as well as dorsoventral patterning (Fig. 6c; Supplementary Data 7), such as cdx447, szl and ved required for DV patterning48,49, and several hox genes50. This suggests that both Adar KD and OE may act on factors with downstream consequence of suppression of these axis-regulating genes.\n",
      "\n",
      "Fig. 6: Differentially expressed genes at 12 hpf in Adar KD and OE. a Principal component analysis of control, Adar knockdown and overexpression samples based on their transcriptome profile. b Number of genes differentially expressed in each condition and their overlap. c Differential expression analysis of Adar KD and OE compared to control. Genes with significant change in expression (p < 0.05) in red. Sequencing read counts were normalized with DESeq2 standard method. Statistical significance is determined by the Wald test corrected for multiple testing using the Benjamini and Hochberg method using DESeq2. Source data: Source data file, Supplementary Data 7. Full size image\n",
      "\n",
      "As Adar gain- and loss-of-function resulted in almost opposite defects in anterior-posterior and dorsoventral patterning, we compared the transcriptome from the two conditions and found 1218 differentially expressed genes (Supplementary Data 7). Genes regulating the development of mesodermal structures (bmpr2a, emilin3a, msgn1, gata1a, meis1b, etv2, pdgfra) as well as hox family genes (hoxa4a/9b, hoxb3a/5a, hoxc1a/3a/6a, hoxd4a) were upregulated in OE compared to KD (Supplementary Data 8). On the contrary, genes regulating epiboly and gastrulation (chuk, epcam, mapkapk2a, snai1a), convergent-extension (dsc2l, gpc4, tdgf1, ptpra, prmt1), and dorsal-ventral patterning (sox11b, ctnnb1, ppp4cb, tll1) were downregulated in OE compared to KD (Supplementary Data 8). These observations agree with the phenotypes resulting from Adar disruption which affected structures along the anteroposterior and dorsoventral axes, where an excess of Adar affected dorsal and anterior structures while Adar deficiency affected more ventral and posterior structures. Taken together, the consequences of Adar disruption at the molecular level could be observed by 12 hpf, which affected the expression of genes involved in the development of various embryonic structures. This suggests that, although no significant changes were observable in global editing levels and transcriptome up to 5.3 hpf, Adar is necessary during the crucial period of embryonic patterning, between gastrulation and 12 hpf.\n",
      "\n",
      "Zygotic Adar is not essential for early embryogenesis\n",
      "\n",
      "We then asked whether zygotic Adar activity is required for later events of embryogenesis. To this end, we created a zebrafish adar mutant line (adar−/−) using the CRISPR/Cas9-based gene knockout method. A 5 bp deletion was introduced into the second exon of the adar gene, resulting in a premature stop codon and polypeptide consisting of 4% (39 aa) of the full length, functional Adar protein (917 aa) (Fig. 7a). To generate homozygous adar mutants we incrossed F1 heterozygous individuals and observed the F2 offspring. Surprisingly, no developmental phenotypes were observed in any of the offspring. We then raised these F2 individuals and genotyped them at ~3 months post-fertilization by fin clipping. Surprisingly, we found 76% heterozygous and 24% wild-type fish among the 200 screened samples. None of the adult F2 individuals were homozygous for the adar−/− mutant allele (Fig. 7b). This suggests that the homozygous adar−/− knockout animals are not viable despite their lack of developmental abnormalities. To determine the fate of the homozygous adar−/− mutants, we genotyped F2 individuals (~300 individuals) at 3 dpf. Interestingly, the distribution of homo-, heterozygous and wild-type individuals was in agreement with the Mendelian ratio (Fig. 7b), indicating that mortality of the homozygous adar−/− individuals occurred later than 3 dpf and before adulthood. In order to pinpoint the timing of mortality, we followed the development of homozygous individuals until the point of death. The adar−/− larvae did not demonstrate any observable morphological abnormalities when compared to heterozygotes or wild-type individuals. Moreover, the swimbladder of adar−/− individuals inflated normally at the same time as that of its wild-type siblings at 5 dpf. However, significant mortality was observed between second- and third weeks post fertilization. Nearly 100% of adar−/− mutants died within this period, with only one surviving until the third month post fertilization, with substantial growth impairment. To eliminate the possibility of genetic compensation51 causing an attenuation of the adar−/− phenotype, we analyzed mRNA levels of adar as well as three other adar family members: adarb1a, adarb1b and adarb2, in wild-type and adar−/− homozygotes. No significant changes in the mRNA levels of adar, adarb1a, adarb1b and adarb2 mRNAs were observed in adar−/− individuals when compared to wild-type at 3 dpf (Fig. 7c). Our findings therefore suggest that the lack of phenotypic alterations in adar−/− larvae, when compared to adar KD (MO-injected) was not due to genetic compensation. It is possible that the late mortality of zygotic adar KO stems from the presence of maternally deposited adar mRNAs and/or proteins from heterozygous mothers, which might be sufficient to drive early developmental processes in the first few days post fertilization. Altogether, these observations suggest that zygotic Adar is not required for early embryogenesis although its function is still essential for life later on.\n",
      "\n",
      "Fig. 7: Zygotic adar−/− mutant exhibit increased innate immune response. a adar−/− mutant generated by CRISPR/Cas9 had a 5-bp deletion within the second exon, resulting in a premature stop codon at 115 bp position. b Different genotype ratios at different time points suggesting that adar−/− homozygotes die between 3 dpf and adult stage. c transcript levels of adar and its paralogs as measured by qRT-PCR at 3 dpf show the lack of genetic compensation effect in homozygous mutant individuals as compared to wild type. Values shown are average of 5 individuals in each WT and adar−/− groups. Data are presented as mean values ± SD. d principal component analysis of wild-type and adar−/− based on their transcriptome profile at 7 dpf. e differential expression analysis revealed a number of significantly upregulated and downregulated genes, which were enriched in functions associated with innate immune response. f Statistical significance for differentially expressed genes is determined by the Wald test corrected for multiple testing using the Benjamini and Hochberg method using DESeq2. Functional enrichment analysis was performed using DAVID which employs Fisher’s Exact test with Benjamini, Bonferroni corrections. Source data: Source data file, Supplementary Data 11, 12. Full size image\n",
      "\n",
      "Loss of zygotic adar cause widespread innate immune response\n",
      "\n",
      "Adar1 loss of function in mammals is characterized by an induction of innate immune response33. In order to test whether this function is conserved in zebrafish, we performed transcriptome analysis on homozygous adar−/− individuals close to the time in which mortality starts to be observed. To this end, we isolated RNA from homozygous adar−/− and its wild-type siblings at 7 dpf and performed RNA-seq. Our analyses revealed that in homozygous adar−/− individuals, 193 and 7 genes were significantly (p val < 0.05) upregulated and downregulated more than fourfold compared to wild-type, respectively (Fig. 7d, Supplementary Data 11). Functional enrichment analysis of overexpressed genes suggest highly significant enrichment of genes related to the innate immune response (Fig. 7e). Notably, genes encoding components of the MDA5-mediated interferon response known to be implicated in mammalian Adar deficiency were found to be highly upregulated in homozygous adar−/− individuals. These include the zebrafish MDA5 orthologue ifih1, as well as the effectors of this pathway irf3 and irf7 52. Another group of interferon-stimulated factors, the IFIT proteins, are known to be transcriptionally induced as a response to viral infection53,54. Strikingly, seven members of this protein family were highly upregulated, including ifit8, ifit9, ifit10, ifit12, ifit14, ifit15, and ifit16. In addition to interferon-response genes, those encoding members of the ubiquitin-proteasome degradation pathway associated with interferon response were found to be upregulated, including psmb8a and psmb13a which are zebrafish orthologues of the interferon inducible subunits β5i and β2i respectively55,56, as well as several genes encoding ubiquitin modifiers and ligases (isg15, usp18, uba7, and CABZ01058333.1 encoding E3 ubiquitin-protein ligase rnf213-alpha-like). Moreover, apoptosis-related genes were also upregulated, including the caspases xaf1, casp21 and casp23.\n",
      "\n",
      "We then searched for specific features of zygotic editing and sought to identify the changes in editing profile caused by the loss of Adar. Interestingly, similar to that observed during embryonic stages, A-to-G conversion were the most enriched among other possible conversions (Supplementary Fig. 6). The editing events at 7 dpf were also prevalent at the 3′-UTR, suggesting that the preference for editing in this region is likely a general feature of Adar activity. Intriguingly, despite the prominent innate immune upregulation observed, we could not see significant alterations in editing pattern between wild-type and adar−/− mutants (Supplementary Fig. 6). Moreover, genes containing editing sites were also not enriched for any particular function, including that of innate immunity (Supplementary Fig. 6, Supplementary Data 12).\n",
      "\n",
      "In summary, the loss of zygotic Adar function in zebrafish induced a widespread innate immune response, proteasome-mediated degradation and apoptosis by 7 dpf, which likely led to death within the second to third weeks post-fertilization. Importantly, the activation of the innate immune response as a result of A-to-I editing deficiency in zebrafish is consistent with the MDA5-dependent Type I interferon response observed in mammalian ADAR deficiency8,34. Our results therefore establish conservation of the zygotic function of Adar between fish and mammals.\n",
      "n\n",
      "Article's Summary:\n",
      "This strongly suggests the presence of A-to-I RNA editing in maternally deposited as well as in zygotic transcripts.\n",
      "The majority of individual reads contained 1–3 RNA editing sites, and never more than five editing sites.\n",
      "Source data: Supplementary Data 1–3, Source data file.\n",
      "Source data: Supplementary Data 4.\n",
      "Source data: Source data file, Supplementary Data 7.\n",
      "n\n"
     ]
    }
   ],
   "source": [
    "url4 =\"https://www.nature.com/articles/s41467-022-33260-6\"\n",
    "toi4_article = Article(url4, language=\"en\")\n",
    "toi4_article.download()\n",
    "toi4_article.parse()\n",
    "toi4_article.nlp()\n",
    "print(\"Article's Title:\")\n",
    "print(toi4_article.title)\n",
    "print(\"n\")\n",
    "print(\"Article's Text:\")\n",
    "print(toi4_article.text)\n",
    "print(\"n\")\n",
    "print(\"Article's Summary:\")\n",
    "print(toi4_article.summary)\n",
    "print(\"n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "https://www.nature.com/ncomms/articles?searchType=journalSearch&sort=PubDate&type=article&page=2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "URL=\"https://www.nature.com/ncomms/articles?searchType=journalSearch&sort=PubDate&type=article&page=\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "ename": "IndexError",
     "evalue": "list index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-22-dd5305b4ed5b>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      9\u001b[0m             \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mf\"{(i-3)+page*15}\"\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mtitles\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     10\u001b[0m         \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 11\u001b[1;33m             \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mf\"{i-3}\"\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mtitles\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtext\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mIndexError\u001b[0m: list index out of range"
     ]
    }
   ],
   "source": [
    "for page in range(1,100):\n",
    "    req = requests.get(URL + str(page) )\n",
    "    soup = bs(req.text, 'html.parser')\n",
    "    \n",
    "    titles = soup.find_all('div',attrs={'class','head'})\n",
    "    \n",
    "    for i in range(5,200):\n",
    "        if page>1:\n",
    "            print(f\"{(i-3)+page*15}\" + titles[i].text)\n",
    "        else:\n",
    "            print(f\"{i-3}\" + titles[i].text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_article_data(url):\n",
    "    response = requests.get(url,headers={\"User-Agent\": \"XY\"})\n",
    "    soup = BeautifulSoup(response.content)\n",
    "    final = \"\"\n",
    "    for tag in soup.find(\"div\",{\"class\":\"td-post-content\"}).find_all(\"p\",recursive=False):\n",
    "        final = final + \" \" + tag.text\n",
    "    return final"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
